// @hitachivantara/uikit-react-core/dist/cjs/components/ColorPicker/ColorPicker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvColorPicker", {
  root: {},
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {
    paddingBottom: "6px",
    display: "block",
    cursor: "pointer"
  },
  description: {},
  headerColorValue: {
    textTransform: "uppercase"
  },
  headerColorIcon: {
    width: 24,
    "& svg": {
      marginLeft: 0
    }
  },
  panel: {
    width: "100%",
    minWidth: uikitStyles.theme.colorPicker.panelMinWidth,
    display: "flex",
    justifyContent: "center",
    padding: uikitStyles.theme.colorPicker.panelPadding
  },
  colorPicker: {
    width: uikitStyles.theme.colorPicker.colorPickerWidth
  },
  colorPickerIcon: {},
  recommendedColorsRoot: {},
  dropdownRootIconOnly: {
    width: 32,
    height: 32
  },
  headerColorIconOnly: {},
  pickerFields: {
    paddingBottom: 20
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
