// @hitachivantara/uikit-react-core/dist/cjs/components/ColorPicker/ColorPicker.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const useControlled = require("../../hooks/useControlled.cjs");
const useUniqueId = require("../../hooks/useUniqueId.cjs");
const useLabels = require("../../hooks/useLabels.cjs");
const useTheme = require("../../hooks/useTheme.cjs");
const setId = require("../../utils/setId.cjs");
const ColorPicker_styles = require("./ColorPicker.styles.cjs");
const PresetColors = require("./PresetColors/PresetColors.cjs");
const Picker = require("./Picker/Picker.cjs");
const SavedColors = require("./SavedColors/SavedColors.cjs");
const FormElement = require("../Forms/FormElement/FormElement.cjs");
const Label = require("../Forms/Label/Label.cjs");
const InfoMessage = require("../Forms/InfoMessage/InfoMessage.cjs");
const BaseDropdown = require("../BaseDropdown/BaseDropdown.cjs");
const Typography = require("../Typography/Typography.cjs");
const Panel = require("../Panel/Panel.cjs");
const DEFAULT_LABELS = {
  recommendedColorsLabel: "Recommended colors:",
  customColorsLabel: "Custom colors:"
};
const HvColorPicker = (props) => {
  const {
    id,
    name,
    required = false,
    disabled = false,
    label,
    "aria-label": ariaLabel,
    "aria-labelledby": ariaLabelledBy,
    description,
    "aria-describedby": ariaDescribedBy,
    className,
    classes: classesProp,
    value,
    onChange,
    onChangeComplete,
    defaultValue = "",
    expanded,
    defaultExpanded = false,
    recommendedColorsPosition = "top",
    recommendedColors = ["#95AFE8", "#E89E5D", "#83B8A6", "#70759C", "#C57E7E", "#FADA95", "#ADBFE8", "#E3B386", "#9AC6B7", "#8B90AF", "#CF9797", "#FAE1AA"],
    showLabels = true,
    labels: labelsProp,
    dropdownIcon = "colorPicker",
    placeholder = "Select color...",
    iconOnly = false,
    showSavedColors = true,
    showCustomColors = true,
    savedColorsValue,
    defaultSavedColorsValue = [],
    onSavedColorAdded,
    onSavedColorRemoved,
    deleteSavedColorButtonArialLabel = "Delete saved color",
    addSavedColorButtonAriaLabel = "Add current color to saved colors"
  } = useDefaultProps.useDefaultProps("HvColorPicker", props);
  const {
    classes,
    css,
    cx
  } = ColorPicker_styles.useClasses(classesProp);
  const {
    activeTheme
  } = useTheme.useTheme();
  const labels = useLabels.useLabels(DEFAULT_LABELS, labelsProp);
  const [isOpen, setIsOpen] = useControlled.useControlled(expanded, Boolean(defaultExpanded));
  const [color, setColor] = useControlled.useControlled(value, defaultValue);
  const [savedColors, setSavedColors] = useControlled.useControlled(savedColorsValue, defaultSavedColorsValue);
  const elementId = useUniqueId.useUniqueId(id, "hvdropdown");
  const hasLabel = label != null;
  const hasDescription = description != null;
  const handleToggle = (_, open) => {
    setIsOpen(open);
  };
  const handleSelect = (val) => {
    onChange == null ? void 0 : onChange(val.hex);
    onChangeComplete == null ? void 0 : onChangeComplete(val.hex);
    setColor(val.hex);
  };
  const handleOnChange = (val) => {
    onChange == null ? void 0 : onChange(val.hex);
    setColor(val.hex);
  };
  const handleOnChangeComplete = (val) => {
    onChangeComplete == null ? void 0 : onChangeComplete(val.hex);
    setColor(val.hex);
  };
  const handleAddColor = () => {
    const colorToAdd = color || "#000000";
    onSavedColorAdded == null ? void 0 : onSavedColorAdded(colorToAdd);
    setSavedColors([...savedColors, colorToAdd]);
  };
  const handleRemoveColor = (val, position) => {
    if (savedColors[position] === val) {
      const sColors = [...savedColors];
      sColors.splice(position, 1);
      onSavedColorRemoved == null ? void 0 : onSavedColorRemoved(val);
      setSavedColors(sColors);
    }
  };
  const setFocusToContent = (containerRef) => {
    const inputs = containerRef == null ? void 0 : containerRef.getElementsByTagName("input");
    if (inputs && inputs.length > 0) {
      inputs[0].focus();
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsxs(FormElement.HvFormElement, { id, name, disabled, required, className: cx(classes.root, className), children: [
    (hasLabel || hasDescription) && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: classes.labelContainer, children: [
      hasLabel && /* @__PURE__ */ jsxRuntime.jsx(Label.HvLabel, { id: setId.setId(elementId, "label"), label, className: classes.label }),
      hasDescription && /* @__PURE__ */ jsxRuntime.jsx(InfoMessage.HvInfoMessage, { id: setId.setId(elementId, "description"), className: classes.description, children: description })
    ] }),
    /* @__PURE__ */ jsxRuntime.jsx(BaseDropdown.HvBaseDropdown, { variableWidth: true, className, expanded: isOpen, onToggle: handleToggle, onContainerCreation: setFocusToContent, classes: {
      root: cx({
        [classes.dropdownRootIconOnly]: iconOnly
      }),
      selection: cx(iconOnly && css({
        padding: 0
      }))
    }, adornment: iconOnly && color ? /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.Checkbox, { className: classes.headerColorIconOnly, color: [color, "transparent"] }) : dropdownIcon === "colorPicker" ? /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.ColorPicker, { className: classes.colorPickerIcon }) : void 0, placeholder: iconOnly ? void 0 : color ? /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
      /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.Checkbox, { className: classes.headerColorIcon, color: [color, "transparent"] }),
      /* @__PURE__ */ jsxRuntime.jsx(Typography.HvTypography, { className: classes.headerColorValue, variant: activeTheme == null ? void 0 : activeTheme.colorPicker.inputValueVariant, children: color })
    ] }) : placeholder, "aria-label": ariaLabel, "aria-labelledby": [label && setId.setId(elementId, "label"), ariaLabelledBy].join(" ").trim() || void 0, "aria-describedby": [description && setId.setId(elementId, "description"), ariaDescribedBy].join(" ").trim() || void 0, children: /* @__PURE__ */ jsxRuntime.jsx(Panel.HvPanel, { className: classes.panel, children: /* @__PURE__ */ jsxRuntime.jsxs("div", { className: classes.colorPicker, children: [
      recommendedColorsPosition === "top" && /* @__PURE__ */ jsxRuntime.jsx(PresetColors.PresetColors, { className: cx((showCustomColors || showSavedColors) && css({
        paddingBottom: activeTheme == null ? void 0 : activeTheme.colorPicker.recommendedColorsBottomPadding
      }), classes.recommendedColorsRoot), colors: recommendedColors, onClick: handleSelect, title: showLabels ? labels == null ? void 0 : labels.recommendedColorsLabel : void 0 }),
      showCustomColors && /* @__PURE__ */ jsxRuntime.jsx(Picker.Picker, { classes: {
        fields: cx({
          [classes.pickerFields]: recommendedColorsPosition === "bottom" || showSavedColors
        })
      }, title: showLabels ? labels == null ? void 0 : labels.customColorsLabel : void 0, color, onChange: handleOnChange, onChangeComplete: handleOnChangeComplete }),
      showSavedColors && /* @__PURE__ */ jsxRuntime.jsx(SavedColors.SavedColors, { colors: savedColors, onAddColor: handleAddColor, onClickColor: handleSelect, onRemoveColor: handleRemoveColor, deleteButtonAriaLabel: deleteSavedColorButtonArialLabel, addButtonAriaLabel: addSavedColorButtonAriaLabel }),
      recommendedColorsPosition === "bottom" && /* @__PURE__ */ jsxRuntime.jsx(PresetColors.PresetColors, { colors: recommendedColors, onClick: handleSelect, title: showLabels ? labels == null ? void 0 : labels.recommendedColorsLabel : void 0 })
    ] }) }) })
  ] });
};
exports.colorPickerClasses = ColorPicker_styles.staticClasses;
exports.HvColorPicker = HvColorPicker;
