// @hitachivantara/uikit-react-core/dist/cjs/components/List/List.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvList", {
  root: {},
  virtualizedRoot: {
    marginBottom: 5
  },
  selectorRoot: {
    width: "100%",
    zIndex: 0
  },
  selectorContainer: {
    "&:hover": {
      backgroundColor: "transparent"
    },
    "&:focus-within": {
      backgroundColor: "transparent",
      outline: "none",
      boxShadow: "none"
    }
  },
  box: {
    width: "32px",
    height: "32px",
    marginLeft: "auto"
  },
  truncate: {
    display: "inline-block",
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  },
  item: {},
  itemSelector: {
    "&:not(:hover):not(.HvIsFocused):not(:focus-within)": {
      backgroundColor: uikitStyles.theme.list.selectedBackgroundColor
    }
  },
  link: {
    ...uikitStyles.theme.typography.body,
    textDecoration: "none",
    "&:focus": {
      boxShadow: "unset !important"
    }
  },
  selectAllSelector: {
    width: "100%",
    margin: "0 0 2px 0",
    position: "relative",
    zIndex: 0,
    // prevent the focus ring to be hidden by sibling hover background
    "&:focus-within": {
      zIndex: 1
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
