// @hitachivantara/uikit-react-core/dist/cjs/components/Badge/Badge.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const labelBaseStyle = {
  ...uikitStyles.theme.typography.label,
  padding: "0 5px",
  color: uikitStyles.theme.colors.atmo1,
  lineHeight: "16px"
};
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvBadge", {
  root: {
    position: "relative",
    "&>*": {
      float: "left"
    }
  },
  badgeContainer: {
    width: 0
  },
  badgePosition: {},
  badge: {
    borderRadius: uikitStyles.theme.space.xs,
    backgroundColor: uikitStyles.theme.colors.secondary,
    float: "left",
    minHeight: "8px",
    minWidth: "8px"
  },
  showCount: {
    ...labelBaseStyle,
    wordBreak: "keep-all"
  },
  showLabel: {
    ...labelBaseStyle,
    wordBreak: "keep-all"
  },
  badgeIcon: {
    position: "relative",
    top: "1px",
    left: "-7px"
  },
  badgeOneDigit: {
    padding: 0,
    width: "16px",
    textAlign: "center"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
