// @hitachivantara/uikit-react-core/dist/cjs/components/Radio/Radio.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvRadio", {
  root: {
    display: "inline-block"
  },
  container: {
    cursor: "pointer",
    display: "flex",
    height: "32px",
    transitionProperty: "background-color",
    transitionDuration: "150ms",
    transitionTimingFunction: "cubic-bezier(0.4, 0, 0.2, 1)",
    transitionDelay: "0ms",
    "&:hover": {
      backgroundColor: uikitStyles.theme.radio.hoverColor,
      borderRadius: uikitStyles.theme.radio.borderRadius
    }
  },
  invalidContainer: {
    borderBottom: `1px solid ${uikitStyles.theme.colors.negative}`,
    "&:hover": {
      borderBottomLeftRadius: "0px",
      borderBottomRightRadius: "0px"
    }
  },
  disabled: {
    cursor: "not-allowed",
    "& $label": {
      color: uikitStyles.theme.colors.secondary_60,
      cursor: "not-allowed"
    }
  },
  radio: {
    height: "32px",
    "& svg": {
      outline: "none",
      boxShadow: "none"
    }
  },
  invalidRadio: {
    borderBottom: `1px solid ${uikitStyles.theme.colors.negative}`,
    "&:hover": {
      borderBottomLeftRadius: "0px",
      borderBottomRightRadius: "0px"
    }
  },
  label: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    verticalAlign: "middle",
    paddingRight: uikitStyles.theme.space.xs,
    whiteSpace: "nowrap",
    ...uikitStyles.theme.typography.body,
    cursor: "pointer",
    height: "32px",
    lineHeight: "32px",
    width: "100%"
  },
  focusVisible: {
    backgroundColor: uikitStyles.theme.colors.atmo3,
    ...focusUtils.outlineStyles
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
