// @hitachivantara/uikit-react-core/dist/cjs/components/TimePicker/TimePicker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  useClasses,
  staticClasses
} = classes.createClasses("HvTimePicker", {
  root: {
    position: "relative"
  },
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {
    marginBottom: 6,
    display: "block"
  },
  description: {},
  placeholder: {
    display: "flex",
    gap: 1
  },
  placeholderDisabled: {
    color: uikitStyles.theme.colors.secondary_60
  },
  dropdownHeader: {},
  dropdownHeaderOpen: {},
  dropdownHeaderInvalid: {
    border: `1px solid ${uikitStyles.theme.colors.negative}`,
    "&:hover": {
      border: `1px solid ${uikitStyles.theme.colors.negative}`
    }
  },
  dropdownPanel: {},
  icon: {},
  timePopperContainer: {
    backgroundColor: uikitStyles.theme.colors.atmo1,
    zIndex: 10,
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    padding: uikitStyles.theme.spacing("xs", 0),
    userSelect: "none",
    minWidth: "175px"
  },
  error: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
