// @hitachivantara/uikit-react-core/dist/cjs/components/Footer/Footer.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvFooter", {
  root: {
    display: "flex",
    alignItems: "center",
    flexDirection: "row",
    padding: `${uikitStyles.theme.space.xs} ${uikitStyles.theme.space.sm}`,
    height: 40,
    bottom: 0,
    zIndex: uikitStyles.theme.zIndices.base,
    backgroundColor: uikitStyles.theme.colors.atmo1,
    boxShadow: `0 -1px 0 ${uikitStyles.theme.colors.atmo4}`
  },
  name: {},
  rightContainer: {
    display: "flex",
    alignItems: "center",
    flexDirection: "row",
    marginLeft: "auto"
  },
  copyright: {},
  separator: {
    width: 1,
    height: 16,
    backgroundColor: uikitStyles.theme.colors.secondary,
    margin: `${uikitStyles.theme.space.xs} ${uikitStyles.theme.space.sm}`
  },
  small: {
    "&$root": {
      height: "unset",
      flexDirection: "column",
      padding: uikitStyles.theme.space.xs
    },
    "& $name": {
      marginBottom: uikitStyles.theme.space.xs
    },
    "& $rightContainer": {
      flexDirection: "column",
      textAlign: "center",
      marginLeft: "unset"
    },
    "& $copyright": {
      flexDirection: "column",
      marginBottom: uikitStyles.theme.space.xs
    },
    "& $separator": {
      display: "none"
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
