// @hitachivantara/uikit-react-core/dist/esm/components/AppSwitcher/Action/Action.js
import { jsx, jsxs, Fragment } from "@emotion/react/jsx-runtime";
import { useState, useCallback } from "react";
import { theme, getColor } from "@hitachivantara/uikit-styles";
import { Info } from "@hitachivantara/uikit-react-icons";
import { useUniqueId } from "../../../hooks/useUniqueId.js";
import TitleWithTooltip from "../TitleWithTooltip.js";
import { useClasses } from "./Action.styles.js";
import { staticClasses } from "./Action.styles.js";
import { HvAvatar } from "../../Avatar/Avatar.js";
import { HvTypography } from "../../Typography/Typography.js";
import { HvListItem } from "../../ListContainer/ListItem/ListItem.js";
import { HvTooltip } from "../../Tooltip/Tooltip.js";
const HvAppSwitcherAction = ({
  id,
  className,
  classes: classesProp,
  application,
  onClickCallback = () => {
  },
  isSelectedCallback = () => false
}) => {
  const {
    classes,
    cx
  } = useClasses(classesProp);
  const {
    name,
    description,
    disabled,
    iconElement,
    iconUrl,
    url,
    target
  } = application;
  const color = disabled ? theme.colors.secondary_60 : getColor(application == null ? void 0 : application.color, theme.colors.secondary);
  const [validIconUrl, setValidIconUrl] = useState(true);
  const renderApplicationIcon = () => {
    if (iconElement) {
      return iconElement;
    }
    if (iconUrl && validIconUrl) {
      return /* @__PURE__ */ jsx("img", { className: classes.iconUrl, src: iconUrl, onError: () => {
        setValidIconUrl(false);
      }, alt: description });
    }
    const brokenTitle = name.split(" ");
    const initials = brokenTitle[0].substring(0, 1) + (brokenTitle[1] ? brokenTitle[1].substring(0, 1) : "");
    return /* @__PURE__ */ jsx(HvAvatar, { size: "sm", backgroundColor: color, variant: "square", "aria-hidden": true, children: initials });
  };
  const isSelected = isSelectedCallback(application);
  const handleOnClick = useCallback((event) => {
    if (disabled) {
      event.preventDefault();
      return;
    }
    onClickCallback == null ? void 0 : onClickCallback(event, {
      ...application,
      isSelected
    });
  }, [application, disabled, isSelected, onClickCallback]);
  const isLink = url != null;
  const descriptionElementId = useUniqueId(id, "hvAction-description");
  const renderApplication = useCallback((children) => {
    const typographyProps = {
      className: classes.typography,
      onClick: handleOnClick,
      style: {
        borderColor: color
      },
      "aria-label": name,
      ...description && {
        "aria-describedby": descriptionElementId
      }
    };
    if (isLink) {
      return /* @__PURE__ */ jsx(HvTypography, { component: "a", href: url, target: target || "_top", ...typographyProps, children });
    }
    return /* @__PURE__ */ jsx(HvTypography, { component: "button", ...typographyProps, children });
  }, [classes.typography, color, description, descriptionElementId, handleOnClick, isLink, name, target, url]);
  return /* @__PURE__ */ jsx(HvListItem, { id, interactive: true, tabIndex: 0, selected: isSelected, disabled, className: cx(classes.root, {
    [classes.disabled]: disabled,
    [classes.selected]: isSelected
  }, className), children: renderApplication(/* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx("div", { className: classes.icon, children: renderApplicationIcon() }),
    /* @__PURE__ */ jsx(TitleWithTooltip, { title: name, className: classes.title }),
    description && /* @__PURE__ */ jsx(HvTooltip, { disableFocusListener: true, disableTouchListener: true, title: /* @__PURE__ */ jsx(HvTypography, { children: description }), children: /* @__PURE__ */ jsx("div", { children: /* @__PURE__ */ jsx(Info, { className: classes.iconInfo, role: "img", "aria-label": description, id: descriptionElementId }) }) })
  ] })) });
};
export {
  HvAppSwitcherAction,
  staticClasses as appSwitcherActionClasses
};
