// @hitachivantara/uikit-react-core/dist/cjs/components/BaseSwitch/BaseSwitch.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvBaseSwitch", {
  root: {
    padding: uikitStyles.theme.baseSwitch.padding,
    cursor: "pointer",
    width: uikitStyles.theme.baseSwitch.width,
    height: uikitStyles.theme.baseSwitch.height,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: uikitStyles.theme.baseSwitch.borderRadius,
    "&:hover": {
      backgroundColor: uikitStyles.theme.baseSwitch.hoverBackgroundColor
    },
    // Higher CSS specificity here
    "& $switchBase": {
      "&:hover": {
        backgroundColor: uikitStyles.theme.baseSwitch.hoverBaseBackgroundColor
      },
      "&$checked": {
        transform: "translateX(16px)",
        "& + $track": {
          opacity: uikitStyles.theme.baseSwitch.checkedOpacity,
          backgroundColor: uikitStyles.theme.baseSwitch.checkedTrackBackgroundColor
        },
        "&:hover": {
          backgroundColor: uikitStyles.theme.baseSwitch.hoverBaseBackgroundColor
        }
      }
    },
    // Higher CSS specificity here
    "&$disabled": {
      "& $switchBase": {
        "& + $track": {
          opacity: uikitStyles.theme.baseSwitch.disabled.trackOpacity,
          backgroundColor: uikitStyles.theme.baseSwitch.disabled.trackBackgroundColor,
          border: uikitStyles.theme.baseSwitch.disabled.trackBorder
        }
      }
    }
  },
  switch: {
    width: uikitStyles.theme.baseSwitch.width,
    height: uikitStyles.theme.baseSwitch.height,
    padding: uikitStyles.theme.baseSwitch.padding,
    "&$readOnly $switchBase": {
      cursor: "default"
    }
  },
  switchBase: {
    width: uikitStyles.theme.baseSwitch.width,
    height: uikitStyles.theme.baseSwitch.height,
    padding: uikitStyles.theme.baseSwitch.padding
  },
  track: {
    opacity: uikitStyles.theme.baseSwitch.track.opacity,
    borderRadius: uikitStyles.theme.baseSwitch.track.borderRadius,
    height: uikitStyles.theme.baseSwitch.track.height,
    width: uikitStyles.theme.baseSwitch.track.width,
    border: uikitStyles.theme.baseSwitch.track.border,
    backgroundColor: uikitStyles.theme.baseSwitch.track.backgroundColor
  },
  thumb: {
    position: "relative",
    left: uikitStyles.theme.baseSwitch.thumb.left,
    width: uikitStyles.theme.baseSwitch.thumb.width,
    height: uikitStyles.theme.baseSwitch.thumb.height,
    border: uikitStyles.theme.baseSwitch.thumb.border,
    backgroundColor: uikitStyles.theme.baseSwitch.thumb.backgroundColor,
    marginLeft: uikitStyles.theme.baseSwitch.thumb.marginLeft,
    marginTop: uikitStyles.theme.baseSwitch.thumb.marginTop,
    boxShadow: uikitStyles.theme.baseSwitch.thumb.boxShadow
  },
  checked: {},
  disabled: {
    cursor: "not-allowed",
    "& $switch": {
      cursor: "not-allowed"
    },
    "& $thumb": {
      backgroundColor: uikitStyles.theme.baseSwitch.disabled.thumbBackgroundColor,
      border: uikitStyles.theme.baseSwitch.disabled.thumbBorder
    }
  },
  readOnly: {},
  focusVisible: {
    borderRadius: uikitStyles.theme.baseSwitch.focusBorderRadius,
    ...focusUtils.outlineStyles
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
