// @hitachivantara/uikit-react-core/dist/cjs/components/ColorPicker/PresetColors/PresetColors.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const name = "HvColorPicker-PresetColors";
const {
  staticClasses,
  useClasses
} = classes.createClasses(name, {
  root: {
    width: uikitStyles.theme.colorPicker.recommendedColorsRootWidth
  },
  colors: {
    display: "flex",
    flexWrap: "wrap",
    position: "relative",
    width: uikitStyles.theme.colorPicker.recommendedColorsWidth,
    margin: uikitStyles.theme.colorPicker.recommendedColorsMargin,
    padding: 0
  },
  title: {
    fontWeight: uikitStyles.theme.fontWeights.semibold,
    marginBottom: 8
  },
  swatchWrap: {
    width: uikitStyles.theme.colorPicker.recommendedColorsSwatchWidth,
    height: uikitStyles.theme.colorPicker.recommendedColorsSwatchHeight,
    margin: uikitStyles.theme.colorPicker.recommendedColorsSwatchMargin,
    "& > span > div": {
      borderRadius: uikitStyles.theme.colorPicker.recommendedColorsSwatchBorderRadius
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
