// @hitachivantara/uikit-react-core/dist/cjs/components/VerticalNavigation/Actions/Action.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../utils/focusUtils.cjs");
const classes = require("../../../utils/classes.cjs");
const hover = () => ({
  background: uikitStyles.theme.verticalNavigation.hoverColor
});
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvVerticalNavigationAction", {
  action: {
    width: "100%",
    display: "flex",
    justifyContent: "flex-start",
    alignItems: "center",
    height: "32px",
    color: uikitStyles.theme.colors.secondary,
    // hover
    "&:hover": hover(),
    "&:focus": {
      outline: "none",
      ...hover()
    },
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    },
    // cursor
    cursor: "pointer",
    "& *": {
      cursor: "pointer"
    }
  },
  noIcon: {
    paddingLeft: uikitStyles.theme.space.xs
  },
  minimized: {
    justifyContent: "center",
    paddingRight: 0
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
