// @hitachivantara/uikit-react-core/dist/cjs/components/RadioGroup/RadioGroup.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvRadioGroup", {
  root: {
    display: "inline-block",
    padding: 0,
    margin: 0,
    overflow: "hidden",
    verticalAlign: "top"
  },
  label: {
    marginBottom: uikitStyles.theme.space.xs
  },
  group: {
    display: "flex"
  },
  vertical: {
    flexDirection: "column",
    // Prevent the focus ring to be hidden by sibling hover background
    "&>*": {
      zIndex: 0
    },
    "&>*:focus-within": {
      zIndex: 1
    }
  },
  horizontal: {
    flexDirection: "row",
    flexWrap: "wrap",
    "&>*:not(:first-of-type)": {
      marginLeft: uikitStyles.theme.space.sm
    },
    width: `calc(100% + ${uikitStyles.theme.space.sm})`
    // Compensate the negative margin left which increases the width
  },
  invalid: {
    paddingBottom: uikitStyles.theme.space.xs,
    borderBottom: `1px solid ${uikitStyles.theme.colors.negative}`
  },
  error: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
