// @hitachivantara/uikit-react-core/dist/cjs/components/VerticalNavigation/VerticalNavigation.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const classes = require("../../utils/classes.cjs");
const uikitStyles = require("@hitachivantara/uikit-styles");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvVerticalNavigation", {
  root: {
    display: "flex",
    flexDirection: "column",
    justifyContent: uikitStyles.theme.verticalNavigation.justifyContent,
    width: "220px",
    background: uikitStyles.theme.colors.atmo1,
    boxShadow: uikitStyles.theme.colors.shadow,
    clipPath: "inset(0px -12px 0px 0px)",
    "& > :only-child": {
      padding: uikitStyles.theme.space.sm
    },
    "& > :not(nav:first-of-type)": {
      borderTop: `3px solid ${uikitStyles.theme.colors.atmo2}`,
      padding: uikitStyles.theme.spacing("xs", "sm", "sm", "sm")
    },
    "& > :first-of-type:not(:last-child)": {
      borderTop: "none",
      padding: uikitStyles.theme.spacing("sm", "sm", "xs", "sm")
    }
  },
  collapsed: {
    width: "56px",
    "&$childData": {
      width: "66px"
    },
    "& > :first-of-type:not(:last-child)": {
      padding: uikitStyles.theme.spacing("sm", "xs", "xs", "xs")
    },
    "& > :not(nav:first-of-type)": {
      padding: uikitStyles.theme.spacing("xs", "xs", "sm", "xs")
    }
  },
  slider: {
    "& > div:first-of-type": {
      borderBottom: `3px solid ${uikitStyles.theme.colors.atmo2}`
    }
  },
  childData: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
