// @hitachivantara/uikit-react-core/dist/esm/components/Carousel/Carousel.js
import { jsxs, jsx } from "@emotion/react/jsx-runtime";
import { useRef, useState, Children, useCallback, useEffect } from "react";
import { useDefaultProps } from "../../hooks/useDefaultProps.js";
import useCarousel from "embla-carousel-react";
import { Close, Fullscreen, Backwards, Forwards } from "@hitachivantara/uikit-react-icons";
import { useTheme } from "../../hooks/useTheme.js";
import { HvCarouselControls } from "./CarouselControls.js";
import { HvCarouselThumbnails } from "./CarouselThumbnails.js";
import { useClasses } from "./Carousel.styles.js";
import { staticClasses } from "./Carousel.styles.js";
import { HvButton } from "../Button/Button.js";
import { HvContainer } from "../Container/Container.js";
import { HvTooltip } from "../Tooltip/Tooltip.js";
import { HvTypography } from "../Typography/Typography.js";
const clamp = (num, max, min = 0) => Math.min(Math.max(num, min), max);
const HvCarousel = (props) => {
  const {
    className,
    classes: classesProp,
    height: heightProp = "auto",
    thumbnailWidth = 90,
    title,
    children,
    actions: actionsProp,
    xs,
    showDots: showDotsProp,
    showCounter: showCounterProp,
    showSlideControls,
    showFullscreen: showFullscreenProp,
    hideThumbnails: hideThumbnailsProp,
    controlsPosition: controlsPositionProp,
    thumbnailsPosition: thumbnailsPositionProp,
    carouselOptions,
    renderThumbnail,
    onChange,
    ...others
  } = useDefaultProps("HvCarousel", props);
  const {
    activeTheme
  } = useTheme();
  const {
    classes,
    css,
    cx
  } = useClasses(classesProp);
  const thumbnailsRef = useRef(null);
  const [isFullscreen, setIsFullscreen] = useState(false);
  const isDs3 = (activeTheme == null ? void 0 : activeTheme.name) === "ds3";
  const actionsPosition = isDs3 ? "header" : "controls";
  const controlsPosition = controlsPositionProp ?? (isDs3 ? "bottom" : "top");
  const thumbnailsPosition = thumbnailsPositionProp ?? "bottom";
  const [containerRef, controller] = useCarousel({
    align: "start",
    loop: true,
    ...carouselOptions
  });
  const [selectedIndex, setSelectedIndex] = useState((carouselOptions == null ? void 0 : carouselOptions.startIndex) ?? 0);
  const numSlides = Children.count(children);
  const handlePrevious = useCallback(() => {
    controller == null ? void 0 : controller.scrollPrev();
  }, [controller]);
  const handleNext = useCallback(() => {
    controller == null ? void 0 : controller.scrollNext();
  }, [controller]);
  const handleScroll = (index) => {
    controller == null ? void 0 : controller.scrollTo(index);
  };
  const handleSelect = useCallback(() => {
    var _a, _b, _c;
    if (!controller)
      return;
    const slideIndex = controller.selectedScrollSnap();
    setSelectedIndex(slideIndex);
    (_c = (_b = (_a = thumbnailsRef.current) == null ? void 0 : _a.querySelectorAll("button")) == null ? void 0 : _b[slideIndex]) == null ? void 0 : _c.scrollIntoView({
      behavior: "smooth",
      block: "nearest"
    });
    onChange == null ? void 0 : onChange(slideIndex);
  }, [controller, onChange]);
  useEffect(() => {
    if (!controller)
      return;
    controller.on("select", handleSelect);
    return () => {
      controller.off("select", handleSelect);
    };
  }, [controller, handleSelect]);
  useEffect(() => {
    if (!controller)
      return;
    controller.reInit();
    setSelectedIndex((currentIndex) => clamp(currentIndex, numSlides, 0));
  }, [numSlides, controller]);
  const canPrev = (controller == null ? void 0 : controller.canScrollPrev()) ?? false;
  const canNext = (controller == null ? void 0 : controller.canScrollNext()) ?? false;
  const showTitle = !!title && (!xs || isFullscreen);
  const showFullscreen = showFullscreenProp ?? xs;
  const height = isFullscreen ? "100%" : heightProp ?? "auto";
  const showCounter = xs;
  const hideThumbnails = hideThumbnailsProp ?? (xs && !isFullscreen);
  const showThumbnails = !hideThumbnails && !!renderThumbnail;
  const showDots = showDotsProp ?? numSlides <= 5;
  const actions = /* @__PURE__ */ jsxs("div", { className: cx(classes.actions, actionsPosition === "header" ? css({
    position: "relative",
    top: -40,
    height: 0
  }) : css({
    position: "absolute"
  })), children: [
    actionsProp,
    showFullscreen && /* @__PURE__ */ jsx(HvTooltip, { title: isFullscreen ? "Close" : "Fullscreen", children: /* @__PURE__ */ jsx(HvButton, { icon: true, onClick: () => setIsFullscreen((curr) => !curr), className: classes.closeButton, children: isFullscreen ? /* @__PURE__ */ jsx(Close, {}) : /* @__PURE__ */ jsx(Fullscreen, {}) }) })
  ] });
  const controls = /* @__PURE__ */ jsx(HvCarouselControls, { classes, showDots, page: selectedIndex, pages: numSlides, canPrevious: canPrev, canNext, onPreviousClick: handlePrevious, onNextClick: handleNext, actions: actionsPosition === "controls" && actions });
  const thumbnails = showThumbnails && /* @__PURE__ */ jsx(HvCarouselThumbnails, { classes, ref: thumbnailsRef, page: selectedIndex, pages: numSlides, width: thumbnailWidth, onThumbnailClick: (evt, i) => handleScroll(i), renderThumbnail });
  return /* @__PURE__ */ jsxs(HvContainer, { className: cx(classes.root, className, {
    [classes.xs]: xs,
    [classes.fullscreen]: isFullscreen
  }), ...others, children: [
    showTitle && /* @__PURE__ */ jsx(HvTypography, { variant: "title2", className: classes.title, children: title }),
    actionsPosition === "header" && actions,
    thumbnailsPosition === "top" && thumbnails,
    controlsPosition === "top" && controls,
    /* @__PURE__ */ jsxs("div", { className: cx(classes.main, {
      [classes.mainXs]: xs,
      [classes.mainFullscreen]: isFullscreen
    }), children: [
      showCounter && /* @__PURE__ */ jsx("div", { className: classes.counterContainer, children: /* @__PURE__ */ jsx("span", { className: classes.counter, children: `${selectedIndex + 1}/${numSlides}` }) }),
      showSlideControls && /* @__PURE__ */ jsxs("div", { className: classes.slideControls, children: [
        /* @__PURE__ */ jsx(HvButton, { icon: true, disabled: !canPrev, variant: "secondarySubtle", "aria-label": "Backwards", onClick: handlePrevious, children: /* @__PURE__ */ jsx(Backwards, { iconSize: "XS" }) }),
        /* @__PURE__ */ jsx(HvButton, { icon: true, disabled: !canNext, variant: "secondarySubtle", "aria-label": "Forwards", onClick: handleNext, children: /* @__PURE__ */ jsx(Forwards, { iconSize: "XS" }) })
      ] }),
      /* @__PURE__ */ jsx("div", { ref: containerRef, style: {
        height
      }, className: classes.slidesViewport, children: /* @__PURE__ */ jsx("div", { className: classes.slidesContainer, children }) })
    ] }),
    controlsPosition === "bottom" && controls,
    thumbnailsPosition === "bottom" && thumbnails
  ] });
};
export {
  HvCarousel,
  staticClasses as carouselClasses
};
