// @hitachivantara/uikit-react-core/dist/cjs/components/DotPagination/DotPagination.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const React = require("react");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const DotPagination_styles = require("./DotPagination.styles.cjs");
const RadioGroup = require("../RadioGroup/RadioGroup.cjs");
const Radio = require("../Radio/Radio.cjs");
const getSelectorIcons = (radioIcon, radioCheckedIcon, classes) => {
  return {
    radio: React.cloneElement(radioIcon || /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.OtherStep, { width: 8, height: 8 }), {
      className: classes == null ? void 0 : classes.icon
    }),
    radioChecked: React.cloneElement(radioCheckedIcon || /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.CurrentStep, { width: 8, height: 8 }), {
      className: classes == null ? void 0 : classes.icon
    })
  };
};
const range = (n) => Array.from(Array(n), (_, i) => i);
const HvDotPagination = (props) => {
  const {
    className,
    classes: classesProp,
    unselectedIcon,
    selectedIcon,
    pages = 1,
    page = 0,
    onPageChange,
    getItemAriaLabel,
    ...others
  } = useDefaultProps.useDefaultProps("HvDotPagination", props);
  const {
    classes,
    cx
  } = DotPagination_styles.useClasses(classesProp);
  const icons = getSelectorIcons(unselectedIcon, selectedIcon, classes);
  return /* @__PURE__ */ jsxRuntime.jsx(RadioGroup.HvRadioGroup, { className: cx(classes.root, className), classes: {
    horizontal: classes.horizontal
  }, orientation: "horizontal", ...others, children: range(pages).map((i) => /* @__PURE__ */ jsxRuntime.jsx(Radio.HvRadio, { classes: {
    radio: classes.radio,
    root: classes.radioRoot
  }, value: i, checked: page === i, onChange: (event) => onPageChange == null ? void 0 : onPageChange(event, i), icon: icons.radio, checkedIcon: icons.radioChecked, "aria-label": getItemAriaLabel == null ? void 0 : getItemAriaLabel(i) }, i)) });
};
exports.dotPaginationClasses = DotPagination_styles.staticClasses;
exports.HvDotPagination = HvDotPagination;
