// @hitachivantara/uikit-react-core/dist/esm/components/TagsInput/TagsInput.styles.js
import { chipClasses } from "@mui/material";
import { theme } from "@hitachivantara/uikit-styles";
import { createClasses } from "../../utils/classes.js";
import { staticClasses as staticClasses$1 } from "../BaseInput/BaseInput.styles.js";
import "../BaseInput/BaseInput.js";
import { staticClasses as staticClasses$2 } from "../Forms/Suggestions/Suggestions.styles.js";
const {
  staticClasses,
  useClasses
} = createClasses("HvTagsInput", {
  input: {
    width: "100%",
    ...theme.typography.caption1
  },
  listItemGutters: {
    padding: "0 5px"
  },
  listItemRoot: {
    marginBottom: 2,
    height: 24,
    lineHeight: "24px",
    "&:not(:last-child)": {
      marginBottom: 2
    },
    "&$singleLine": {
      display: "table-cell",
      paddingTop: "2px"
    }
  },
  root: {
    display: "inline-block",
    width: "100%"
  },
  chipRoot: {
    [`&.${chipClasses.root}`]: {
      maxWidth: "none"
    }
  },
  disabled: {
    "& $tagsList": {
      backgroundColor: theme.tagsInput.disabledBackgroundColor,
      border: `1px solid ${theme.colors.atmo4}`,
      "&:focus-within, &:hover": {
        border: `1px solid ${theme.colors.atmo4}`
      }
    }
  },
  readOnly: {
    "& $tagsList": {
      backgroundColor: theme.tagsInput.readOnlyBackgroundColor,
      border: `1px solid ${theme.tagsInput.readOnlyBorderColor}`,
      "&:hover": {
        border: `1px solid ${theme.tagsInput.readOnlyBorderColor}`
      }
    }
  },
  resizable: {
    width: "auto",
    resize: "both",
    overflow: "auto"
  },
  invalid: {},
  labelContainer: {
    float: "left",
    display: "flex",
    alignItems: "flex-start"
  },
  label: {
    display: "block",
    float: "left",
    paddingBottom: "6px"
  },
  description: {
    display: "block",
    float: "left"
  },
  characterCounter: {
    display: "block",
    float: "right",
    textAlign: "right",
    marginBottom: "6px"
  },
  tagsList: {
    display: "flex",
    alignContent: "flex-start",
    float: "left",
    clear: "both",
    width: "100%",
    maxWidth: "100%",
    height: "32px",
    padding: 5,
    overflow: "auto",
    position: "relative",
    flexDirection: "row",
    flexWrap: "wrap",
    backgroundColor: theme.colors.atmo1,
    border: `1px solid ${theme.tagsInput.borderColor}`,
    borderRadius: theme.radii.base,
    "&:hover": {
      cursor: "text",
      border: `1px solid ${theme.tagsInput.hoverColor}`
    },
    [`& .${staticClasses$1.inputRoot}`]: {
      border: "none"
    },
    "&:focus, &:focus-within, &:focus-visible": {
      outlineColor: "#52A8EC",
      outlineStyle: "solid",
      outlineWidth: "0px",
      outlineOffset: "-1px",
      boxShadow: "0 0 0 1px #52A8EC, 0 0 0 4px rgba(29,155,209,.3)"
    },
    "&$singleLine": {
      overflowX: "hidden",
      overflowY: "hidden",
      height: 32,
      display: "table-row",
      paddingTop: 0
    },
    "&$error": {
      border: `1px solid ${theme.colors.negative}`
    },
    "&$invalid": {
      border: `1px solid ${theme.colors.negative}`
    }
  },
  tagInputContainerRoot: {
    display: "flex",
    flexGrow: 1,
    height: 24,
    lineHeight: "24px",
    "&$singleLine": {
      display: "table-cell",
      minWidth: 60,
      width: "100%",
      paddingTop: "3px",
      verticalAlign: "middle"
    }
  },
  tagInputRoot: {
    [`& .${staticClasses$1.root}`]: {
      width: "100%",
      border: "none",
      "&:hover $tagInputBorderContainer": {
        background: "none"
      }
    },
    [`&& .${staticClasses$1.inputRoot}`]: {
      marginLeft: 0,
      marginRight: 0,
      flex: "1 1 auto",
      height: 24,
      lineHeight: "24px",
      padding: 0,
      border: "none"
    },
    [`& .${staticClasses$1.inputBorderContainer}`]: {
      border: "none",
      background: "none"
    },
    [`& .${staticClasses$1.inputRootFocused}`]: {
      outline: "none",
      boxShadow: "none"
    },
    [`& .${staticClasses$1.root} .${staticClasses$1.inputRootReadOnly}`]: {
      backgroundColor: "transparent",
      border: "none",
      "&:hover": {
        border: "none"
      }
    },
    [`&& .${staticClasses$1.input}`]: {
      marginLeft: 0
    }
  },
  tagSelected: {
    outlineColor: "#52A8EC",
    outlineStyle: "solid",
    outlineWidth: "0px",
    outlineOffset: "-1px",
    boxShadow: "0 0 0 1px #52A8EC, 0 0 0 4px rgba(29,155,209,.3)"
  },
  tagInputBorderContainer: {},
  tagInputRootFocused: {},
  tagInputRootEmpty: {},
  singleLine: {},
  error: {
    float: "left"
  },
  inputExtension: {},
  suggestionsContainer: {
    width: "100%",
    position: "relative",
    top: 59,
    [`& .${staticClasses$2.root} .${staticClasses$2.list} &`]: {
      width: "100%"
    }
  },
  suggestionList: {}
});
export {
  staticClasses,
  useClasses
};
