// @hitachivantara/uikit-react-core/dist/esm/components/FilterGroup/FilterContent/FilterContent.js
import { jsxs, Fragment, jsx } from "@emotion/react/jsx-runtime";
import { useState, useContext, useRef, useMemo } from "react";
import { Filters } from "@hitachivantara/uikit-react-icons";
import { useTheme } from "../../../hooks/useTheme.js";
import { setId } from "../../../utils/setId.js";
import { useClasses } from "./FilterContent.styles.js";
import { staticClasses } from "./FilterContent.styles.js";
import { HvFilterGroupContext } from "../FilterGroupContext.js";
import { HvFilterGroupLeftPanel } from "../LeftPanel/LeftPanel.js";
import { HvFilterGroupRightPanel } from "../RightPanel/RightPanel.js";
import { HvButton } from "../../Button/Button.js";
import { HvTypography } from "../../Typography/Typography.js";
import { HvBaseDropdown } from "../../BaseDropdown/BaseDropdown.js";
import { HvFilterGroupCounter } from "../Counter/Counter.js";
import { HvActionBar } from "../../ActionBar/ActionBar.js";
const HvFilterGroupContent = ({
  id,
  status,
  disabled = false,
  "aria-label": ariaLabel,
  "aria-labelledby": ariaLabelledBy,
  description,
  "aria-describedby": ariaDescribedBy,
  onChange,
  onCancel,
  onClear,
  labels,
  horizontalPlacement = "right",
  disablePortal = true,
  escapeWithReference = true,
  height,
  leftEmptyElement,
  rightEmptyElement,
  classes: classesProp,
  ...others
}) => {
  const {
    classes
  } = useClasses(classesProp);
  const {
    activeTheme
  } = useTheme();
  const [filterGroupOpen, setFilterGroupOpen] = useState(false);
  const {
    defaultValue,
    filterValues,
    rollbackFilters,
    clearFilters,
    applyFilters,
    applyDisabled
  } = useContext(HvFilterGroupContext);
  const focusTarget = useRef(null);
  const focusOnContainer = () => {
    var _a;
    (_a = focusTarget.current) == null ? void 0 : _a.focus();
  };
  const onApplyHandler = (event) => {
    applyFilters();
    onChange == null ? void 0 : onChange(event, filterValues);
    setFilterGroupOpen(false);
  };
  const onCancelHandler = (event) => {
    rollbackFilters();
    onCancel == null ? void 0 : onCancel(event);
    setFilterGroupOpen(false);
  };
  const onClearHandler = (event) => {
    clearFilters();
    onClear == null ? void 0 : onClear(event);
  };
  const handleToggle = (event, open) => {
    if (event === null)
      return;
    setFilterGroupOpen(open);
    if (!open)
      onCancelHandler == null ? void 0 : onCancelHandler(event);
  };
  const Header = useMemo(() => /* @__PURE__ */ jsxs(Fragment, { children: [
    /* @__PURE__ */ jsx(Filters, {}),
    /* @__PURE__ */ jsx(HvTypography, { variant: "label", children: labels == null ? void 0 : labels.placeholder })
  ] }), [labels == null ? void 0 : labels.placeholder]);
  return /* @__PURE__ */ jsxs(HvBaseDropdown, { id: setId(id, "dropdown"), role: "combobox", classes: {
    root: classes.dropdown,
    panel: classes.panel,
    selection: classes.baseDropdownSelection,
    header: classes.header
  }, disabled, disablePortal, variableWidth: true, placement: horizontalPlacement, expanded: filterGroupOpen, onToggle: handleToggle, onClickOutside: onCancelHandler, onContainerCreation: focusOnContainer, placeholder: Header, adornment: /* @__PURE__ */ jsx(HvFilterGroupCounter, {}), popperProps: {
    modifiers: [{
      name: "preventOverflow",
      enabled: escapeWithReference
    }]
  }, "aria-haspopup": "dialog", "aria-label": ariaLabel, "aria-labelledby": ariaLabelledBy, "aria-invalid": status === "invalid" ? true : void 0, "aria-errormessage": status === "invalid" ? setId(id, "error") : void 0, "aria-describedby": [description && setId(id, "description"), ariaDescribedBy].join(" ").trim() || void 0, ...others, children: [
    /* @__PURE__ */ jsx("div", { ref: focusTarget, tabIndex: -1 }),
    /* @__PURE__ */ jsxs("div", { className: classes.root, style: {
      height
    }, children: [
      /* @__PURE__ */ jsx(HvFilterGroupLeftPanel, { id, className: classes.leftSidePanel, emptyElement: leftEmptyElement }),
      /* @__PURE__ */ jsx(HvFilterGroupRightPanel, { id, className: classes.rightSidePanel, emptyElement: rightEmptyElement, labels })
    ] }),
    /* @__PURE__ */ jsxs(HvActionBar, { className: classes.actionBar, children: [
      /* @__PURE__ */ jsx(HvButton, { id: setId(id, "clearFilters-button"), disabled: defaultValue ? (defaultValue == null ? void 0 : defaultValue.flat().length) === (filterValues == null ? void 0 : filterValues.flat().length) : (filterValues == null ? void 0 : filterValues.flat().length) === 0, variant: "secondaryGhost", onClick: onClearHandler, children: labels == null ? void 0 : labels.clearLabel }),
      /* @__PURE__ */ jsx("div", { "aria-hidden": "true", className: classes.space, children: " " }),
      /* @__PURE__ */ jsx(HvButton, { id: setId(id, "apply-button"), disabled: applyDisabled, variant: activeTheme == null ? void 0 : activeTheme.filterGroup.applyButtonVariant, onClick: onApplyHandler, className: classes.applyButton, children: labels == null ? void 0 : labels.applyLabel }),
      /* @__PURE__ */ jsx(HvButton, { id: setId(id, "cancel-button"), variant: activeTheme == null ? void 0 : activeTheme.filterGroup.cancelButtonVariant, onClick: onCancelHandler, children: labels == null ? void 0 : labels.cancelLabel })
    ] })
  ] });
};
export {
  HvFilterGroupContent,
  staticClasses as filterGroupContentClasses
};
