// @hitachivantara/uikit-react-core/dist/cjs/components/Snackbar/Snackbar.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const material = require("@mui/material");
const classes = require("../../utils/classes.cjs");
const {
  useClasses,
  staticClasses
} = classes.createClasses("HvSnackbar", {
  root: {},
  anchorOriginTopRight: {
    [`&.${material.snackbarClasses.anchorOriginTopRight}`]: {
      top: uikitStyles.theme.space.xs,
      right: uikitStyles.theme.space.xs
    }
  },
  anchorOriginTopLeft: {
    [`&.${material.snackbarClasses.anchorOriginTopLeft}`]: {
      top: uikitStyles.theme.space.xs,
      left: uikitStyles.theme.space.xs
    }
  },
  anchorOriginTopCenter: {
    [`&.${material.snackbarClasses.anchorOriginTopCenter}`]: {
      top: uikitStyles.theme.space.xs
    }
  },
  anchorOriginBottomCenter: {
    [`&.${material.snackbarClasses.anchorOriginBottomCenter}`]: {
      bottom: uikitStyles.theme.space.xs
    }
  },
  anchorOriginBottomLeft: {
    [`&.${material.snackbarClasses.anchorOriginBottomLeft}`]: {
      bottom: uikitStyles.theme.space.xs,
      left: uikitStyles.theme.space.xs
    }
  },
  anchorOriginBottomRight: {
    [`&.${material.snackbarClasses.anchorOriginBottomRight}`]: {
      bottom: uikitStyles.theme.space.xs,
      right: uikitStyles.theme.space.xs
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
