// @hitachivantara/uikit-react-core/dist/cjs/components/Tooltip/Tooltip.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const material = require("@mui/material");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvTooltip", {
  root: {},
  tooltip: {},
  tooltipMulti: {},
  popper: {},
  title: {},
  valuesContainer: {},
  values: {},
  color: {},
  separatorColor: {},
  separator: {},
  valueWrapper: {}
});
const popperSx = (useSingle) => {
  return {
    [`& .${material.tooltipClasses.popper}`]: {
      opacity: 1
    },
    [`& .${material.tooltipClasses.tooltip}`]: {
      fontFamily: uikitStyles.theme.fontFamily.body,
      ...uikitStyles.theme.typography.body,
      display: "flex",
      width: "fit-content",
      maxWidth: 532,
      backgroundColor: uikitStyles.theme.colors.atmo1,
      boxShadow: uikitStyles.theme.colors.shadow,
      padding: 0,
      ...useSingle && {
        padding: "15px 20px",
        borderRadius: uikitStyles.theme.tooltip.borderRadius
      },
      "& p": {
        display: "-webkit-box",
        width: "fit-content",
        boxOrient: "vertical",
        textOverflow: "ellipsis",
        overflow: "hidden",
        ...useSingle && {
          wordBreak: "break-word"
        }
      }
    },
    [`& .${staticClasses.title}`]: {
      padding: "15px 20px",
      borderBottom: `3px solid ${uikitStyles.theme.colors.atmo2}`
    },
    [`& .${staticClasses.valuesContainer}`]: {
      padding: uikitStyles.theme.space.sm
    },
    [`& .${staticClasses.values}`]: {
      display: "flex",
      justifyContent: "space-between",
      paddingBottom: "10px",
      "&:last-child": {
        paddingBottom: 0
      }
    },
    [`& .${staticClasses.color}`]: {
      width: uikitStyles.theme.space.xs
    },
    [`& .${staticClasses.separator}`]: {
      width: uikitStyles.theme.space.xs
    },
    [`& .${staticClasses.separatorColor}`]: {
      width: "5px"
    },
    [`& .${staticClasses.valueWrapper}`]: {
      padding: uikitStyles.theme.space.sm
    }
  };
};
exports.popperSx = popperSx;
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
