// @hitachivantara/uikit-react-core/dist/cjs/components/Header/Navigation/MenuItem/MenuItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../../utils/focusUtils.cjs");
const classes = require("../../../../utils/classes.cjs");
const item = {
  border: "none",
  cursor: "pointer",
  padding: `${uikitStyles.theme.space.xs} ${uikitStyles.theme.space.sm}`,
  height: "100%",
  display: "flex",
  alignItems: "center",
  "&:active": {
    outline: "none"
  },
  "&:focus": {
    outline: "none"
  },
  "&:focus-visible": {
    ...focusUtils.outlineStyles
  },
  "& span": {
    display: "inline-flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-between"
  },
  "& span::after": {
    content: "attr(data-text)",
    height: 0,
    visibility: "hidden",
    overflow: "hidden",
    userSelect: "none",
    pointerEvents: "none",
    fontWeight: uikitStyles.theme.fontWeights.semibold,
    "@media speech": {
      display: "none"
    }
  }
};
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvHeader-MenuItem", {
  root: {
    display: "inline",
    "& span": {
      color: uikitStyles.theme.header.color
    },
    "&:hover": {
      backgroundColor: uikitStyles.theme.header.hoverColor
    },
    "&:focus-within": {
      backgroundColor: uikitStyles.theme.header.hoverColor
    }
  },
  menu: {
    marginTop: uikitStyles.theme.header.secondLevelSelectedItemBorderTopThickness,
    paddingBottom: uikitStyles.theme.header.secondLevelSelectedItemBorderBottomThickness,
    paddingTop: uikitStyles.theme.header.secondLevelSelectedItemBorderBottomThickness,
    height: "100%",
    borderTop: `${uikitStyles.theme.header.secondLevelSelectedItemBorderTopThickness} solid ${uikitStyles.theme.header.secondLevelBackgroundColor}`,
    borderBottom: `${uikitStyles.theme.header.secondLevelSelectedItemBorderBottomThickness} solid ${uikitStyles.theme.header.secondLevelBackgroundColor}`,
    "&& > * > span": {
      color: uikitStyles.theme.header.secondLevelItemColor,
      padding: uikitStyles.theme.header.secondLevelItemPadding
    }
  },
  menubar: {
    marginTop: uikitStyles.theme.header.selectedItemBorderTopThickness,
    paddingBottom: uikitStyles.theme.header.selectedItemBorderBottomThickness,
    paddingTop: uikitStyles.theme.header.selectedItemBorderBottomThickness,
    height: "100%",
    borderTop: `${uikitStyles.theme.header.selectedItemBorderTopThickness} solid ${uikitStyles.theme.header.backgroundColor}`,
    borderBottom: `${uikitStyles.theme.header.selectedItemBorderBottomThickness} solid ${uikitStyles.theme.header.backgroundColor}`,
    "& > * > span": {
      padding: uikitStyles.theme.header.itemPadding
    }
  },
  selected: {
    "&$menu": {
      marginTop: 0,
      paddingBottom: 0,
      borderTop: `${uikitStyles.theme.header.secondLevelSelectedItemBorderTopThickness} solid ${uikitStyles.theme.header.secondLevelSelectedItemBorderTopColor}`,
      borderBottom: `${uikitStyles.theme.header.secondLevelSelectedItemBorderBottomThickness} solid ${uikitStyles.theme.header.secondLevelSelectedItemBorderBottomColor}`,
      paddingTop: uikitStyles.theme.header.secondLevelSelectedItemBorderBottomThickness,
      height: "100%",
      "& > * > span": {
        color: uikitStyles.theme.header.secondLevelSelectedItemColor,
        backgroundColor: uikitStyles.theme.header.secondLevelSelectedItemBackgroundColor,
        borderRadius: uikitStyles.theme.header.secondLevelSelectedItemBorderRadius
      }
    },
    "&$menubar": {
      marginTop: 0,
      paddingBottom: 0,
      borderTop: `${uikitStyles.theme.header.selectedItemBorderTopThickness} solid ${uikitStyles.theme.header.selectedItemBorderTopColor}`,
      borderBottom: `${uikitStyles.theme.header.selectedItemBorderBottomThickness} solid ${uikitStyles.theme.header.selectedItemBorderBottomColor}`,
      paddingTop: uikitStyles.theme.header.selectedItemBorderBottomThickness,
      height: "100%",
      "& > * > span": {
        color: uikitStyles.theme.header.selectedItemColor,
        backgroundColor: uikitStyles.theme.header.selectedItemBackgroundColor,
        borderRadius: uikitStyles.theme.header.selectedItemBorderRadius
      }
    }
  },
  link: {
    textDecoration: "none",
    ...item
  },
  button: {
    color: "inherit",
    ...item
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
