// @hitachivantara/uikit-react-core/dist/esm/components/Card/Header/Header.js
import { jsx } from "@emotion/react/jsx-runtime";
import MuiCardHeader from "@mui/material/CardHeader";
import { useTheme } from "../../../hooks/useTheme.js";
import { useClasses } from "./Header.styles.js";
import { staticClasses } from "./Header.styles.js";
const HvCardHeader = ({
  classes: classesProp,
  className,
  title,
  subheader,
  icon,
  onClick,
  ...others
}) => {
  const {
    activeTheme
  } = useTheme();
  const {
    classes,
    css,
    cx
  } = useClasses(classesProp);
  return /* @__PURE__ */ jsx(MuiCardHeader, { title, subheader, action: icon, onClick, className: cx(classes.root, className), classes: {
    title: cx(css({
      ...activeTheme == null ? void 0 : activeTheme.typography[activeTheme == null ? void 0 : activeTheme.card.titleVariant]
    }), {
      [classes.titleShort]: icon,
      [classes.title]: !icon
    }),
    subheader: cx(css({
      ...activeTheme == null ? void 0 : activeTheme.typography[activeTheme == null ? void 0 : activeTheme.card.subheaderVariant],
      color: activeTheme == null ? void 0 : activeTheme.card.subheaderColor
    }), classes.subheader),
    action: classes.action,
    content: classes.content
  }, ...others });
};
export {
  HvCardHeader,
  staticClasses as cardHeaderClasses
};
