// @hitachivantara/uikit-react-core/dist/cjs/components/FileUploader/File/File.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const classes = require("../../../utils/classes.cjs");
const uikitStyles = require("@hitachivantara/uikit-styles");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvFile", {
  root: {},
  progressbar: {
    position: "absolute",
    top: "-1px"
  },
  progressbarContainer: {
    height: uikitStyles.theme.fileUploader.file.progressHeight
  },
  progressbarBack: {},
  nameText: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  },
  progressTextContainer: {
    display: "flex",
    flexGrow: 1,
    alignItems: "center"
  },
  removeButton: {
    width: 32,
    height: 32,
    margin: `0px ${uikitStyles.theme.space.xs}`
  },
  previewContainer: {
    display: "flex",
    margin: `0px ${uikitStyles.theme.space.xs}`,
    width: uikitStyles.theme.fileUploader.file.previewContainerSize,
    height: uikitStyles.theme.fileUploader.file.previewContainerSize,
    justifyContent: "center",
    alignItems: "center",
    "& span": {
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      width: "100%",
      height: "100%"
    },
    "& img": {
      width: uikitStyles.theme.fileUploader.file.imageSize,
      height: uikitStyles.theme.fileUploader.file.imageSize,
      objectFit: "cover",
      objectPosition: "center",
      alignSelf: "center"
    }
  },
  icon: {
    width: 32,
    height: 32,
    margin: `0px ${uikitStyles.theme.space.xs}`
  },
  fail: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
