// @hitachivantara/uikit-react-core/dist/cjs/components/Tag/Tag.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const hexToRgbA = require("../../utils/hexToRgbA.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvTag", {
  root: {},
  chipRoot: {
    "&.MuiChip-root": {
      height: 16,
      borderRadius: 0,
      maxWidth: 180,
      fontFamily: uikitStyles.theme.fontFamily.body,
      "&:focus-visible": {
        backgroundColor: hexToRgbA.hexToRgbA(uikitStyles.theme.colors.base_light, 0.3)
      },
      "&$categorical": {
        borderRadius: 8,
        "& $label": {
          color: uikitStyles.theme.colors.secondary
        }
      }
    },
    "&$disabled": {
      opacity: 1,
      backgroundColor: uikitStyles.theme.colors.atmo3,
      "& $label": {
        color: uikitStyles.theme.colors.secondary_60
      }
    },
    "& .MuiChip-label": {
      paddingLeft: uikitStyles.theme.space.xs,
      paddingRight: uikitStyles.theme.space.xs,
      ...uikitStyles.theme.typography.caption1,
      color: uikitStyles.theme.colors.base_dark,
      "& p": {
        color: uikitStyles.theme.colors.base_dark
      }
    },
    "& .MuiChip-deleteIcon": {
      marginRight: 0,
      width: 16,
      height: 16,
      padding: 0,
      "&:hover": {
        backgroundColor: uikitStyles.theme.button.hoverColor
      },
      "&:focus": {
        ...focusUtils.outlineStyles,
        borderRadius: 0
      },
      "&:focus:not(:focus-visible)": {
        outline: "0 !important",
        boxShadow: "none !important"
      }
    }
  },
  disabled: {},
  clickable: {
    cursor: "pointer",
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    }
  },
  categorical: {},
  label: {},
  deleteIcon: {},
  // TODO: redundant - use deleteIcon. remove in v6
  button: {},
  tagButton: {},
  // TODO: redundant - use $clickable or :not($disabled). remove in v6
  focusVisible: {},
  disabledDeleteIcon: {},
  categoricalFocus: {},
  categoricalDisabled: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
