// @hitachivantara/uikit-react-core/dist/cjs/components/Carousel/Carousel.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvCarousel", {
  /** Styles applied to the component root class. */
  root: {
    display: "flex",
    alignItems: "stretch",
    flexDirection: "column",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    margin: 0,
    paddingTop: uikitStyles.theme.spacing(2),
    paddingBottom: uikitStyles.theme.spacing(2)
  },
  xs: {
    "&:not(._)": {
      padding: 0
    },
    // put dots on top of Slide
    "& $dots": {
      position: "relative",
      top: "-40px"
    },
    "& $actions": {
      top: 0
    },
    "& $controls": {
      display: uikitStyles.theme.carousel.xsControlsDisplay
    }
  },
  fullscreen: {
    width: "100vw",
    height: "100vh",
    position: "fixed",
    inset: 0,
    zIndex: uikitStyles.theme.zIndices.modal,
    "&:not(._)": {
      padding: uikitStyles.theme.spacing("xs", "xl")
    }
  },
  title: {
    display: "flex",
    justifyContent: "flex-start",
    marginBottom: uikitStyles.theme.space.xs
  },
  actions: {
    display: "flex",
    justifyContent: "flex-end",
    right: 0,
    position: "relative"
  },
  closeButton: {},
  mainContainer: {},
  controls: {
    display: "flex",
    alignItems: "center",
    position: "relative",
    height: 32,
    justifyContent: uikitStyles.theme.carousel.controlsJustifyContent,
    backgroundColor: uikitStyles.theme.carousel.controlsBackgroundColor,
    border: uikitStyles.theme.carousel.controlsBorder,
    gap: uikitStyles.theme.space.xs,
    "&:has($dots)": {
      justifyContent: "center"
    }
  },
  pageCounter: {},
  main: {
    padding: 0,
    display: "flex",
    flexDirection: "column",
    position: "relative",
    "&:hover $slideControls": {
      opacity: 1
    }
  },
  mainXs: {},
  mainFullscreen: {
    maxHeight: "80vh"
  },
  counterContainer: {
    position: "absolute",
    top: uikitStyles.theme.space.xs,
    right: uikitStyles.theme.space.md,
    zIndex: 1,
    display: uikitStyles.theme.carousel.counterContainerDisplay
  },
  counter: {
    color: uikitStyles.theme.colors.base_light,
    backgroundColor: uikitStyles.theme.colors.base_dark,
    padding: uikitStyles.theme.spacing(0, "sm")
  },
  slideControls: {
    position: "absolute",
    left: 0,
    right: 0,
    top: `calc(50% - (32px / 2))`,
    padding: uikitStyles.theme.spacing(0, "sm"),
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    opacity: 0,
    "& button": {
      zIndex: 1
    },
    "&:focus-within": {
      opacity: 1
    }
  },
  slidesViewport: {
    overflow: "hidden"
  },
  slidesContainer: {
    display: "flex",
    flexDirection: "row",
    height: "100%"
  },
  dots: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    gap: uikitStyles.theme.space.xs,
    height: uikitStyles.theme.space.md
  },
  dotsXs: {},
  dot: {
    width: 5,
    height: 5,
    margin: uikitStyles.theme.space.xs,
    borderRadius: "50%",
    backgroundColor: uikitStyles.theme.colors.atmo4
  },
  dotSelected: {
    width: 10,
    height: 10,
    backgroundColor: uikitStyles.theme.colors.secondary_80
  },
  panel: {
    display: "flex",
    width: "100%",
    overflowX: "auto",
    // "hidden",
    overflowY: "hidden",
    padding: uikitStyles.theme.spacing("xs", "2px", "2px")
  },
  thumbnail: {
    height: "unset",
    padding: 0,
    "& img": {
      width: "100%",
      height: "100%",
      textAlign: "center",
      aspectRatio: "16/9",
      opacity: "50%",
      borderRadius: uikitStyles.theme.carousel.thumbnailBorderRadius
    }
  },
  thumbnailSelected: {
    "& img": {
      border: uikitStyles.theme.carousel.thumbnailSelectedBorder,
      opacity: "100%"
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
