// @hitachivantara/uikit-react-core/dist/cjs/components/DatePicker/DatePicker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvDatePicker", {
  root: {
    position: "relative"
  },
  actionContainer: {
    justifyContent: "space-between"
  },
  leftContainer: {},
  rightContainer: {},
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {
    paddingBottom: "6px",
    display: "block"
  },
  description: {},
  error: {},
  dropdown: {
    display: "block"
  },
  panel: {},
  dropdownHeaderInvalid: {
    border: `1px solid ${uikitStyles.theme.colors.negative}`,
    "&:hover": {
      border: `1px solid ${uikitStyles.theme.colors.negative}`
    }
  },
  dropdownHeaderOpen: {},
  icon: {},
  action: {
    "&:first-of-type": {
      marginRight: uikitStyles.theme.space.xs
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
