// @hitachivantara/uikit-react-core/dist/cjs/components/Accordion/Accordion.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvAccordion", {
  root: {
    "& + root": {
      paddingTop: 8
    }
  },
  hidden: {
    height: 0,
    display: "none"
  },
  container: {
    paddingTop: 8,
    height: "auto"
  },
  label: {
    width: "100%",
    display: "flex",
    justifyContent: "flex-start",
    alignItems: "center",
    height: "32px",
    "&[disabled], &:active": {
      outline: "none"
    },
    "&:focus": {
      outline: "none",
      background: uikitStyles.theme.colors.atmo3
    },
    "&:hover": {
      background: uikitStyles.theme.colors.atmo3
    },
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    },
    cursor: "pointer"
  },
  disabled: {
    cursor: "not-allowed",
    color: uikitStyles.theme.colors.secondary_60,
    "&:focus": {
      background: "none"
    },
    "&:hover": {
      background: "none"
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
