// @hitachivantara/uikit-react-core/dist/esm/components/InlineEditor/InlineEditor.styles.js
import { theme } from "@hitachivantara/uikit-styles";
import { createClasses } from "../../utils/classes.js";
import { staticClasses as staticClasses$1 } from "../BaseInput/BaseInput.styles.js";
import { staticClasses as staticClasses$2 } from "../Input/Input.styles.js";
const {
  staticClasses,
  useClasses
} = createClasses("HvInlineEditor", {
  root: {
    [`& .${staticClasses$1.inputRoot}.${staticClasses$2.inputRoot}`]: {
      height: "100%",
      minHeight: "32px"
    }
  },
  inputBorderContainer: {
    top: "unset",
    bottom: 0
  },
  input: {},
  inputRoot: {},
  text: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    alignSelf: "center"
  },
  largeText: {},
  textEmpty: {
    color: theme.colors.secondary_60
  },
  button: {
    padding: theme.spacing("6px", "8px", "5px", "8px"),
    minHeight: "32px",
    boxSizing: "border-box",
    cursor: "text",
    height: "100%",
    width: "100%",
    maxWidth: "100%",
    justifyContent: "flex-start",
    alignItems: "center",
    backgroundColor: "transparent",
    border: `${theme.inlineEditor.borderWidth} solid transparent`,
    "&:hover, &:focus": {
      border: `${theme.inlineEditor.borderWidth} solid ${theme.inlineEditor.hoverBorderColor}`,
      backgroundColor: "transparent",
      "& $icon": {
        visibility: "visible"
      }
    },
    "&:active": {
      border: `${theme.inlineEditor.borderWidth} solid ${theme.inlineEditor.activeBorderColor}`,
      backgroundColor: "transparent",
      "& $icon": {
        visibility: "visible"
      }
    },
    "& > div": {
      width: "100%"
    },
    "& > div > span": {
      width: "100%"
    }
  },
  icon: {
    cursor: "pointer",
    visibility: "hidden",
    alignSelf: "center",
    height: "16px",
    width: "32px",
    minWidth: "32px",
    "& svg": {
      margin: theme.spacing(0, "xs")
    }
  },
  iconVisible: {
    visibility: "visible"
  }
});
export {
  staticClasses,
  useClasses
};
