// @hitachivantara/uikit-react-core/dist/cjs/components/ColorPicker/Picker/Picker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const name = "HvColorPicker-Picker";
const {
  staticClasses,
  useClasses
} = classes.createClasses(name, {
  pickers: {
    display: "flex",
    flexDirection: uikitStyles.theme.colorPicker.pickersFlexDirection
  },
  saturation: {
    width: uikitStyles.theme.colorPicker.saturationWidth,
    height: uikitStyles.theme.colorPicker.saturationHeight,
    marginRight: uikitStyles.theme.colorPicker.saturationMarginRight,
    position: "relative",
    overflow: "visible",
    "& > div": {
      borderRadius: uikitStyles.theme.colorPicker.saturationBorderRadius,
      "& .saturation-white": {
        borderRadius: uikitStyles.theme.colorPicker.saturationBorderRadius,
        "& .saturation-black": {
          borderRadius: uikitStyles.theme.colorPicker.saturationBorderRadius
        }
      }
    }
  },
  saturationPointer: {
    width: uikitStyles.theme.colorPicker.saturationPointerWidth,
    height: uikitStyles.theme.colorPicker.saturationPointerHeight,
    boxShadow: `0 0 0 2px #fff, inset 0 0 1px 1px rgba(0,0,0,.3),
            0 0 1px 2px rgba(0,0,0,.4)`,
    borderRadius: "50%",
    transform: "translate(-3px, -3px)"
  },
  hue: {
    height: uikitStyles.theme.colorPicker.hueHeight,
    width: uikitStyles.theme.colorPicker.hueWidth,
    position: "relative",
    overflow: "visible",
    marginTop: uikitStyles.theme.colorPicker.hueMarginTop,
    "& .hue-horizontal": {
      borderRadius: uikitStyles.theme.colorPicker.hueBorderRadius
    }
  },
  hueSlider: {
    width: uikitStyles.theme.colorPicker.hueSliderWidth,
    height: uikitStyles.theme.colorPicker.hueSliderHeight,
    background: uikitStyles.theme.colorPicker.hueSliderBackground,
    boxShadow: "0 0 2px rgb(0 0 0 / 60%)",
    marginLeft: uikitStyles.theme.colorPicker.hueSliderMarginLeft,
    border: uikitStyles.theme.colorPicker.hueSliderBorder,
    borderRadius: uikitStyles.theme.colorPicker.hueSliderBorderRadius,
    transform: "translate(0, -2px)"
  },
  title: {
    fontWeight: uikitStyles.theme.fontWeights.semibold,
    marginBottom: 8
  },
  fields: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
