// @hitachivantara/app-shell/dist/esm/components/layout/Header/Header.js
import { jsxs as f, jsx as o } from "@emotion/react/jsx-runtime";
import { Helmet as g } from "react-helmet-async";
import { useTranslation as v } from "react-i18next";
import { HvHeader as h, HvHeaderBrand as N, HvHeaderNavigation as H } from "@hitachivantara/uikit-react-core";
import C from "./styles.js";
import u from "../../../lib/hooks/useAppShellConfig.js";
import I from "../../../lib/hooks/useNavigation.js";
import M from "../../../providers/hooks/useNavigationContext.js";
import { CONFIG_TRANSLATIONS_NAMESPACE as A } from "../../../lib/i18n/index.js";
import O from "./HeaderActions/HeaderActions.js";
import S from "./BrandLogo/BrandLogo.js";
const T = () => {
  const {
    t: a
  } = v(A), e = u(), {
    items: n,
    selectedMenuItemId: m,
    rootMenuItemId: s,
    isCompactMode: d
  } = M(), {
    navigate: l
  } = I(), i = e.navigationMode === "ONLY_TOP", p = !d && e.navigationMode !== "ONLY_LEFT" && n.length > 0, c = (x, t) => {
    t.href && l(t.href, {
      state: {
        selectedItemId: t.id
      }
    });
  }, r = e.name ? a(e.name) : "";
  return /* @__PURE__ */ f(h, { position: "fixed", children: [
    /* @__PURE__ */ o(g, { children: /* @__PURE__ */ o("title", { children: r }) }),
    /* @__PURE__ */ o(N, { logo: /* @__PURE__ */ o(C, { children: /* @__PURE__ */ o(S, { logo: e.logo }) }), name: r }),
    p && /* @__PURE__ */ o(H, { data: n, selected: i ? m : s, onClick: c, levels: i ? 2 : 1 }),
    /* @__PURE__ */ o(O, {})
  ] });
}, G = T;
export {
  G as default
};
