// @hitachivantara/uikit-react-core/dist/cjs/components/FilterGroup/FilterContent/FilterContent.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const name = "HvFilterGroupContent";
const {
  staticClasses,
  useClasses
} = classes.createClasses(name, {
  dropdown: {},
  panel: {
    maxHeight: 500,
    minHeight: 370
  },
  header: {
    display: "flex",
    justifyContent: "space-between",
    height: 32
  },
  baseDropdownSelection: {
    padding: uikitStyles.theme.spacing("0px", "30px", "0px", "0px")
  },
  root: {
    width: 640,
    maxHeight: "calc(500px - 75px)",
    minHeight: "calc(370px - 75px)"
  },
  rightSidePanel: {
    display: "inline-block",
    width: "50%",
    height: "100%",
    maxHeight: "calc(500px - 75px)",
    minHeight: "calc(370px - 75px)",
    verticalAlign: "top",
    overflow: "visible",
    boxShadow: uikitStyles.theme.filterGroup.rightPanelShadow,
    borderLeft: uikitStyles.theme.filterGroup.rightPanelBorderLeft
  },
  leftSidePanel: {
    display: "inline-block",
    width: `calc(50% - ${uikitStyles.theme.space.sm} - ${uikitStyles.theme.space.sm} + 8px)`,
    height: `calc(100% - ${uikitStyles.theme.space.sm} - ${uikitStyles.theme.space.sm} + 8px)`,
    verticalAlign: "top",
    maxHeight: "calc(500px - 75px)",
    minHeight: "calc(370px - 75px)",
    padding: 4,
    margin: `calc(${uikitStyles.theme.spacing("sm")} - 4px)`
  },
  actionBar: {},
  space: {
    flex: 1
  },
  applyButton: {
    marginRight: uikitStyles.theme.filterGroup.applyButtonMarginRight
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
