// @hitachivantara/uikit-react-core/dist/cjs/components/Button/Button.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvButton", {
  root: {
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "center",
    textTransform: "none",
    cursor: "pointer",
    minWidth: "70px",
    whiteSpace: "nowrap",
    "&:hover": {},
    "&:focus": {},
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    },
    "&:active": {},
    // default button - no size specified
    fontFamily: uikitStyles.theme.fontFamily.body,
    fontSize: uikitStyles.theme.fontSizes.base,
    fontWeight: uikitStyles.theme.fontWeights.semibold,
    lineHeight: "11px",
    letterSpacing: 0,
    height: "32px",
    borderRadius: uikitStyles.theme.button.borderRadius,
    padding: uikitStyles.theme.button.padding
  },
  startIcon: {
    marginLeft: `calc(-1 * ${uikitStyles.theme.space.xs})`
  },
  endIcon: {
    marginRight: `calc(-1 * ${uikitStyles.theme.space.xs})`
  },
  focusVisible: {},
  disabled: {
    color: uikitStyles.theme.colors.secondary_60,
    borderColor: uikitStyles.theme.colors.atmo4,
    backgroundColor: uikitStyles.theme.colors.atmo3,
    cursor: "not-allowed",
    pointerEvents: "none"
  },
  icon: {
    margin: 0,
    padding: 0,
    height: "fit-content",
    minWidth: "unset"
  }
});
const getVariantStyles = (variant) => {
  const variantStyles = {
    primary: {
      color: uikitStyles.theme.colors.atmo1,
      backgroundColor: uikitStyles.theme.colors.primary,
      "&:hover, &:focus-visible": {
        backgroundColor: uikitStyles.theme.colors.primary_80
      }
    },
    primarySubtle: {
      color: uikitStyles.theme.colors.primary,
      backgroundColor: "transparent",
      border: `1px solid ${uikitStyles.theme.colors.primary}`,
      "&:hover, &:focus-visible": {
        backgroundColor: uikitStyles.theme.button.hoverColor
      }
    },
    primaryGhost: {
      color: uikitStyles.theme.colors.primary,
      backgroundColor: "transparent",
      "&:hover, &:focus-visible": {
        backgroundColor: uikitStyles.theme.button.hoverColor
      },
      "&:disabled": {
        backgroundColor: "transparent"
      }
    },
    secondarySubtle: {
      color: uikitStyles.theme.colors.secondary,
      backgroundColor: uikitStyles.theme.button.secondaryBackgroundColor,
      border: `1px solid ${uikitStyles.theme.button.secondarySubtleBorderColor}`,
      "&:hover, &:focus-visible": {
        backgroundColor: uikitStyles.theme.button.hoverColor
      }
    },
    secondaryGhost: {
      color: uikitStyles.theme.colors.secondary,
      backgroundColor: "transparent",
      "&:hover, &:focus-visible": {
        backgroundColor: uikitStyles.theme.button.hoverColor
      },
      "&:disabled": {
        backgroundColor: "transparent"
      }
    },
    semantic: {
      color: uikitStyles.theme.colors.base_dark,
      backgroundColor: "transparent",
      "&:hover, &:focus-visible": {
        backgroundColor: uikitStyles.theme.button.semanticColor
      },
      "&:disabled": {
        backgroundColor: uikitStyles.theme.button.semanticColorDisabled
      }
    },
    secondary: void 0,
    ghost: void 0
  };
  return variantStyles[variant];
};
const getRadiusStyles = (radius) => ({
  borderRadius: uikitStyles.theme.radii[radius]
});
const getSizeStyles = (size) => ({
  height: uikitStyles.theme.sizes[size],
  paddingLeft: uikitStyles.theme.space[size],
  paddingRight: uikitStyles.theme.space[size],
  fontSize: uikitStyles.theme.fontSizes[size]
});
const getOverrideColors = () => ({
  "& svg .color0": {
    fill: "currentcolor"
  }
});
exports.getOverrideColors = getOverrideColors;
exports.getRadiusStyles = getRadiusStyles;
exports.getSizeStyles = getSizeStyles;
exports.getVariantStyles = getVariantStyles;
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
