// @hitachivantara/uikit-react-core/dist/esm/components/Banner/Banner.js
import { jsx } from "@emotion/react/jsx-runtime";
import { useCallback } from "react";
import Slide from "@mui/material/Slide";
import Snackbar from "@mui/material/Snackbar";
import { setId } from "../../utils/setId.js";
import { useDefaultProps } from "../../hooks/useDefaultProps.js";
import { useClasses } from "./Banner.styles.js";
import { staticClasses } from "./Banner.styles.js";
import { HvBannerContent } from "./BannerContent/BannerContent.js";
const HvBanner = (props) => {
  const {
    id,
    classes: classesProp,
    className,
    open,
    onClose,
    anchorOrigin = "top",
    variant = "default",
    transitionDuration = 300,
    transitionDirection = "down",
    showIcon = false,
    customIcon,
    actions,
    actionsCallback,
    actionsPosition = "auto",
    label,
    offset = 60,
    bannerContentProps,
    ...others
  } = useDefaultProps("HvBanner", props);
  const {
    classes,
    cx
  } = useClasses(classesProp);
  const anchorOriginBanner = {
    horizontal: "center",
    vertical: anchorOrigin
  };
  const SlideTransition = useCallback((properties) => /* @__PURE__ */ jsx(Slide, { ...properties, direction: transitionDirection }), [transitionDirection]);
  return /* @__PURE__ */ jsx(Snackbar, { id, open, className, classes: {
    root: cx(classes.root, {
      [classes.rootClosed]: !open
    }),
    anchorOriginTopCenter: classes.anchorOriginTopCenter,
    anchorOriginBottomCenter: classes.anchorOriginBottomCenter
  }, style: {
    [anchorOrigin]: offset
  }, anchorOrigin: anchorOriginBanner, TransitionComponent: SlideTransition, transitionDuration, ...others, children: /* @__PURE__ */ jsx(HvBannerContent, { id: setId(id, "content"), content: label, variant, customIcon, showIcon, actions, actionsCallback, actionsPosition, onClose, ...bannerContentProps }) });
};
export {
  HvBanner,
  staticClasses as bannerClasses
};
