// @hitachivantara/uikit-react-core/dist/cjs/components/FilterGroup/RightPanel/RightPanel.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const cloneDeep = require("lodash/cloneDeep");
const setId = require("../../../utils/setId.cjs");
const FilterGroupContext = require("../FilterGroupContext.cjs");
const RightPanel_styles = require("./RightPanel.styles.cjs");
const Typography = require("../../Typography/Typography.cjs");
const CheckBox = require("../../CheckBox/CheckBox.cjs");
const Panel = require("../../Panel/Panel.cjs");
const Input = require("../../Input/Input.cjs");
const List = require("../../List/List.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const cloneDeep__default = /* @__PURE__ */ _interopDefault(cloneDeep);
const HvFilterGroupRightPanel = ({
  id,
  className,
  labels,
  emptyElement,
  classes: classesProp
}) => {
  const {
    classes
  } = RightPanel_styles.useClasses(classesProp);
  const [searchStr, setSearchStr] = React.useState("");
  const [allSelected, setAllSelected] = React.useState(false);
  const [anySelected, setAnySelected] = React.useState(false);
  const {
    filterOptions,
    filterValues = [],
    setFilterValues,
    activeGroup
  } = React.useContext(FilterGroupContext.HvFilterGroupContext);
  const activeGroupOptions = React.useMemo(() => {
    var _a;
    return ((_a = filterOptions[activeGroup]) == null ? void 0 : _a.data.map((option) => option.id)) || [];
  }, [filterOptions, activeGroup]);
  const activeFilterValues = React.useMemo(() => {
    var _a;
    return ((_a = filterValues[activeGroup]) == null ? void 0 : _a.filter((value) => activeGroupOptions.includes(value))) || [];
  }, [filterValues, activeGroupOptions, activeGroup]);
  const listValues = React.useMemo(() => {
    var _a;
    return ((_a = filterOptions[activeGroup]) == null ? void 0 : _a.data.map((option) => {
      var _a2;
      return {
        ...option,
        label: option.name,
        selected: (_a2 = filterValues[activeGroup]) == null ? void 0 : _a2.includes(option.id),
        isHidden: option.name.toLowerCase().indexOf(searchStr.toLowerCase()) < 0
      };
    })) || [];
  }, [filterOptions, filterValues, activeGroup, searchStr]);
  const updateSelectAll = React.useCallback(() => {
    const nbrSelected = activeFilterValues == null ? void 0 : activeFilterValues.length;
    const hasSelection = nbrSelected > 0;
    const allSelect = nbrSelected === activeGroupOptions.length;
    setAnySelected(hasSelection);
    setAllSelected(hasSelection && allSelect);
  }, [activeFilterValues, activeGroupOptions]);
  React.useEffect(() => {
    updateSelectAll();
  }, [activeFilterValues, updateSelectAll]);
  React.useEffect(() => setSearchStr(""), [activeGroup]);
  const onChangeHandler = (values) => {
    const newFilterValues = filterOptions.map((_, i) => activeGroup === i ? values.filter((v) => v.selected).map((v) => v.id) : [...filterValues[i] || []]);
    setFilterValues(newFilterValues);
  };
  const handleSelectAll = React.useCallback(() => {
    const newFilterValues = cloneDeep__default.default(filterValues);
    newFilterValues[activeGroup] = anySelected ? [] : activeGroupOptions;
    setFilterValues(newFilterValues);
  }, [activeGroup, activeGroupOptions, anySelected, filterValues, setFilterValues]);
  const SelectAll = React.useCallback(() => {
    const nbrSelected = activeFilterValues == null ? void 0 : activeFilterValues.length;
    const defaultLabel = /* @__PURE__ */ jsxRuntime.jsx(Typography.HvTypography, { component: "span", children: nbrSelected > 0 ? /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
      /* @__PURE__ */ jsxRuntime.jsx("b", { children: nbrSelected }),
      ` ${labels == null ? void 0 : labels.multiSelectionConjunction} ${activeGroupOptions.length}`
    ] }) : /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
      /* @__PURE__ */ jsxRuntime.jsx("b", { children: labels == null ? void 0 : labels.selectAll }),
      ` (${activeGroupOptions.length})`
    ] }) });
    return /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.selectAllContainer, children: /* @__PURE__ */ jsxRuntime.jsx(CheckBox.HvCheckBox, { id: setId.setId(id, "select-all"), label: defaultLabel, onChange: () => handleSelectAll(), className: classes.selectAll, indeterminate: anySelected && !allSelected, checked: allSelected }) });
  }, [activeFilterValues == null ? void 0 : activeFilterValues.length, activeGroupOptions.length, allSelected, anySelected, handleSelectAll, id, labels, classes == null ? void 0 : classes.selectAllContainer, classes == null ? void 0 : classes.selectAll]);
  return /* @__PURE__ */ jsxRuntime.jsx(Panel.HvPanel, { id: setId.setId(id, "rightPanel"), className, children: listValues.length > 0 ? /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsx(Input.HvInput, { id: setId.setId(id, "search"), classes: {
      root: classes.search
    }, type: "search", placeholder: labels == null ? void 0 : labels.searchBoxPlaceholder, value: searchStr, onChange: (_, str) => setSearchStr(str) }),
    /* @__PURE__ */ jsxRuntime.jsx(SelectAll, {}),
    /* @__PURE__ */ jsxRuntime.jsx(List.HvList, { id: setId.setId(id, "list"), values: listValues, className: classes.list, multiSelect: true, useSelector: true, showSelectAll: false, onChange: onChangeHandler, selectable: true, condensed: true, hasTooltips: true }, activeGroup)
  ] }) : emptyElement });
};
exports.filterGroupRightPanelClasses = RightPanel_styles.staticClasses;
exports.HvFilterGroupRightPanel = HvFilterGroupRightPanel;
