// @hitachivantara/uikit-react-core/dist/cjs/components/Dropdown/Dropdown.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvDropdown", {
  root: {
    width: "100%",
    position: "relative",
    display: "inline-block"
  },
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {
    paddingBottom: "6px",
    display: "block"
  },
  description: {},
  error: {},
  placeholder: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  },
  selectionDisabled: {
    lineHeight: uikitStyles.theme.space.md,
    color: uikitStyles.theme.dropdown.disabledColor
  },
  dropdown: {
    width: "100%",
    borderRadius: uikitStyles.theme.radii.base,
    "& $dropdownHeaderInvalid": {
      border: uikitStyles.theme.dropdown.dropdownHeaderInvalidBorder,
      "&:hover": {
        border: uikitStyles.theme.dropdown.dropdownHeaderInvalidBorder
      }
    }
  },
  arrow: {},
  dropdownHeader: {},
  dropdownHeaderInvalid: {},
  dropdownHeaderOpen: {},
  dropdownListContainer: {},
  rootList: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
