// @hitachivantara/uikit-react-core/dist/cjs/components/ColorPicker/Fields/Fields.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const name = "HvColorPicker-Fields";
const {
  staticClasses,
  useClasses
} = classes.createClasses(name, {
  fields: {
    width: "100%",
    display: "flex",
    paddingTop: uikitStyles.theme.colorPicker.fieldsPaddingTop,
    marginRight: uikitStyles.theme.colorPicker.fieldsMarginRight
  },
  single: {
    maxWidth: uikitStyles.theme.colorPicker.fieldsRgbWidth,
    paddingLeft: uikitStyles.theme.colorPicker.fieldsRgbPaddingLeft,
    "& input": {
      marginLeft: 5,
      marginRight: 5
    },
    "& label": {
      paddingLeft: 5
    }
  },
  double: {
    maxWidth: uikitStyles.theme.colorPicker.fieldsHexWidth,
    paddingRight: uikitStyles.theme.colorPicker.fieldsHexPaddingRight,
    "& input": {
      textTransform: "uppercase",
      marginLeft: 5,
      marginRight: 5
    },
    "& label": {
      paddingLeft: 5
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
