// @hitachivantara/uikit-react-core/dist/cjs/components/Container/Container.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const uikitStyles = require("@hitachivantara/uikit-styles");
const styles = require("@mui/material/styles");
const MuiContainer = require("@mui/material/Container");
const useMediaQuery = require("@mui/material/useMediaQuery");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const Container_styles = require("./Container.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const MuiContainer__default = /* @__PURE__ */ _interopDefault(MuiContainer);
const useMediaQuery__default = /* @__PURE__ */ _interopDefault(useMediaQuery);
const HvContainer = React.forwardRef((props, ref) => {
  const {
    maxWidth = false,
    classes: classesProp,
    className,
    fixed,
    disableGutters,
    ...others
  } = useDefaultProps.useDefaultProps("HvContainer", props);
  const {
    classes,
    cx,
    css
  } = Container_styles.useClasses(classesProp);
  const muiTheme = styles.useTheme();
  const upMd = useMediaQuery__default.default(muiTheme.breakpoints.up("md"));
  const gutters = upMd ? {
    // Increases specificity
    [`&.${Container_styles.staticClasses.root}`]: {
      paddingLeft: uikitStyles.theme.spacing(4),
      paddingRight: uikitStyles.theme.spacing(4)
    }
  } : {
    // Increases specificity
    [`&.${Container_styles.staticClasses.root}`]: {
      paddingLeft: uikitStyles.theme.space.sm,
      paddingRight: uikitStyles.theme.space.sm
    }
  };
  return /* @__PURE__ */ jsxRuntime.jsx(MuiContainer__default.default, { className: cx({
    [css(gutters)]: !disableGutters
  }, className), classes: {
    root: classes.root,
    disableGutters: classes.disableGutters,
    fixed: classes.fixed,
    maxWidthXs: classes.maxWidthXs,
    maxWidthSm: classes.maxWidthSm,
    maxWidthMd: classes.maxWidthMd,
    maxWidthLg: classes.maxWidthLg,
    maxWidthXl: classes.maxWidthXl
  }, ref, maxWidth, fixed, disableGutters, ...others });
});
exports.containerClasses = Container_styles.staticClasses;
exports.HvContainer = HvContainer;
