// @hitachivantara/app-shell/dist/esm/components/layout/Main/Main.js
import { jsxs as c, jsx as t } from "@emotion/react/jsx-runtime";
import { HvContainer as d } from "@hitachivantara/uikit-react-core";
import p from "../../../providers/hooks/useNavigationContext.js";
import { StyledContainer as l, StyledMain as f } from "./styles.js";
import h from "../../../providers/hooks/useLayoutContext.js";
import x from "../../../lib/hooks/useAppShellConfig.js";
import g from "../VerticalNavigation/VerticalNavigation.js";
const u = ({
  children: a
}) => {
  const {
    hasVerticalNavigation: r,
    verticalNavigationMode: e,
    showHeaderSubMenu: n,
    isCompactMode: i,
    switchVerticalNavigationMode: m
  } = p(), {
    bannerMaxHeight: s
  } = h(), {
    mainPanel: o
  } = x();
  return /* @__PURE__ */ c(l, { showHeaderSubMenu: n, isCompactMode: i, children: [
    r && e !== "CLOSED" && /* @__PURE__ */ t(g, { onClickAway: i ? m : void 0 }),
    /* @__PURE__ */ t(f, { paddingTop: s, children: /* @__PURE__ */ t(d, { maxWidth: (o == null ? void 0 : o.maxWidth) ?? "xl", children: a }) })
  ] });
}, H = u;
export {
  H as default
};
