// @hitachivantara/uikit-react-core/dist/cjs/components/Calendar/SingleCalendar/CalendarCell.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../utils/focusUtils.cjs");
const classes = require("../../../utils/classes.cjs");
const hover = {
  backgroundColor: uikitStyles.theme.calendar.cellHoverColor,
  cursor: "pointer"
};
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvCalendarCell", {
  cellContainer: {
    cursor: "pointer",
    border: 0,
    padding: 0,
    margin: 0,
    backgroundColor: "transparent"
  },
  focusSelection: {
    "&:hover": {
      ...hover
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      ...hover,
      ...focusUtils.outlineStyles
    }
  },
  calendarDate: {
    display: "flex",
    justifyContent: "center",
    flexDirection: "column",
    textAlign: "center",
    height: "40px",
    width: "40px"
  },
  calendarDateSelected: {
    backgroundColor: uikitStyles.theme.colors.atmo3,
    color: uikitStyles.theme.colors.secondary,
    "&:hover": {
      ...hover
    }
  },
  calendarDateNotInMonth: {
    color: uikitStyles.theme.colors.secondary_60,
    cursor: "not-allowed"
  },
  calendarDateInSelectionRange: {
    backgroundColor: uikitStyles.theme.colors.atmo3
  },
  calendarDateDisabled: {
    color: uikitStyles.theme.colors.secondary_60,
    cursor: "no-drop"
  },
  startBookend: {
    borderLeft: `1px solid ${uikitStyles.theme.colors.secondary}`
  },
  endBookend: {
    borderRight: `1px solid ${uikitStyles.theme.colors.secondary}`
  },
  dateWrapper: {
    width: "40px",
    height: "40px"
  },
  cellsInRange: {
    "&:focus": {
      outside: "none"
    }
  },
  cellsOutsideRange: {
    "&:focus": {
      outside: "none"
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
