// @hitachivantara/uikit-react-core/dist/cjs/components/Card/Card.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvCard", {
  root: {
    overflow: "visible",
    position: "relative",
    outline: uikitStyles.theme.card.outline,
    borderRadius: uikitStyles.theme.card.borderRadius,
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    },
    "&:focus": {
      outline: "none"
    }
  },
  selected: {
    outline: `1px solid ${uikitStyles.theme.colors.secondary}`,
    "&:hover": {
      outline: `1px solid ${uikitStyles.theme.colors.secondary}`
    },
    "&:focus": {
      outline: `1px solid ${uikitStyles.theme.colors.secondary}`
    }
  },
  selectable: {
    "&:hover": {
      outline: `1px solid ${uikitStyles.theme.card.hoverColor}`
    }
  },
  semanticContainer: {
    position: "relative",
    "& > *": {
      position: "absolute",
      zIndex: 1
    }
  },
  icon: {
    top: `calc(${uikitStyles.theme.card.iconMargin} + ${uikitStyles.theme.space.xs})`,
    right: `calc(${uikitStyles.theme.card.iconMargin} + ${uikitStyles.theme.space.xs})`
  },
  semanticBar: {
    width: "100%",
    top: -1,
    right: 0
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
