// @hitachivantara/uikit-react-core/dist/esm/components/Tooltip/Tooltip.js
import { jsx } from "@emotion/react/jsx-runtime";
import { Tooltip, Fade } from "@mui/material";
import { forwardRef } from "react";
import { useTheme } from "../../hooks/useTheme.js";
import { useDefaultProps } from "../../hooks/useDefaultProps.js";
import { useClasses, popperSx } from "./Tooltip.styles.js";
import { staticClasses } from "./Tooltip.styles.js";
const HvTooltip = forwardRef((props, ref) => {
  const {
    className,
    classes: classesProp,
    open,
    enterDelay = 300,
    placement = "top",
    useSingle = true,
    children,
    title,
    TransitionComponent = Fade,
    TransitionProps = {
      timeout: 400,
      placement
    },
    containerId,
    ...others
  } = useDefaultProps("HvTooltip", props);
  const {
    rootId
  } = useTheme();
  const {
    classes
  } = useClasses(classesProp);
  return /* @__PURE__ */ jsx(Tooltip, { ref, open: open ?? void 0, enterDelay, placement, TransitionComponent, TransitionProps, className, classes: {
    tooltip: useSingle ? classes.tooltip : classes.tooltipMulti,
    popper: classes.popper
  }, title, PopperProps: {
    sx: popperSx(useSingle),
    container: typeof window !== "undefined" ? document.getElementById(containerId || rootId || "") || document.body : void 0
  }, ...others, children });
});
export {
  HvTooltip,
  staticClasses as tooltipClasses
};
