// @hitachivantara/uikit-react-core/dist/cjs/components/Tab/Tab.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const classes = require("../../utils/classes.cjs");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvTab", {
  root: {
    marginTop: "3px",
    padding: uikitStyles.theme.tab.padding,
    minWidth: 70,
    minHeight: 32,
    textTransform: "none",
    fontFamily: uikitStyles.theme.fontFamily.body,
    ...uikitStyles.theme.typography.body,
    "&:hover": {
      backgroundColor: uikitStyles.theme.tab.hoverBackgroundColor,
      borderRadius: uikitStyles.theme.tab.hoverBackgroundBorderRadius,
      "&::after": {
        height: "1px",
        backgroundColor: uikitStyles.theme.tab.hoverUnderlineBackgroundColor
      }
    },
    "&$selected": {
      color: uikitStyles.theme.typography.label.color,
      fontWeight: uikitStyles.theme.typography.label.fontWeight,
      lineHeight: uikitStyles.theme.typography.label.lineHeight,
      letterSpacing: uikitStyles.theme.typography.label.letterSpacing
    },
    "&$disabled": {
      color: uikitStyles.theme.colors.secondary_60,
      cursor: "not-allowed",
      pointerEvents: "all",
      opacity: 1,
      "&:hover": {
        background: "none"
      }
    },
    opacity: 1,
    "&::after": {
      position: "absolute",
      left: 0,
      top: "calc(100% - 1px)",
      height: "1px",
      width: "100%",
      backgroundColor: uikitStyles.theme.colors.atmo4,
      content: "''"
    },
    // Override Mui styling: https://mui.com/material-ui/api/tab/#css
    "& .MuiTab-iconWrapper": {
      margin: 0
    }
  },
  focusVisible: {
    ...focusUtils.outlineStyles
  },
  selected: {},
  disabled: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
