// @hitachivantara/uikit-react-core/dist/cjs/components/Forms/FormElement/FormElement.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const useDefaultProps = require("../../../hooks/useDefaultProps.cjs");
const useUniqueId = require("../../../hooks/useUniqueId.cjs");
const FormUtils = require("./utils/FormUtils.cjs");
const FormElementContext = require("./context/FormElementContext.cjs");
const FormElementValueContext = require("./context/FormElementValueContext.cjs");
const FormElementDescriptorsContext = require("./context/FormElementDescriptorsContext.cjs");
const FormElement_styles = require("./FormElement.styles.cjs");
const HvFormElement = (props) => {
  const {
    classes: classesProp,
    className,
    children,
    id,
    name,
    value,
    disabled = false,
    required = false,
    readOnly = false,
    status = "standBy",
    ...others
  } = useDefaultProps.useDefaultProps("HvFormElement", props);
  const {
    classes,
    cx
  } = FormElement_styles.useClasses(classesProp);
  const elementId = useUniqueId.useUniqueId(id, "hvformelement");
  const contextValue = React.useMemo(() => ({
    elementId,
    elementName: name,
    elementStatus: status,
    elementDisabled: disabled,
    elementRequired: required,
    elementReadOnly: readOnly
  }), [disabled, elementId, name, readOnly, required, status]);
  const descriptors = React.useMemo(() => FormUtils.findDescriptors(children), [children]);
  return /* @__PURE__ */ jsxRuntime.jsx("div", { id, className: cx(classes.root, className), ...others, children: /* @__PURE__ */ jsxRuntime.jsx(FormElementContext.HvFormElementContextProvider, { value: contextValue, children: /* @__PURE__ */ jsxRuntime.jsx(FormElementValueContext.HvFormElementValueContextProvider, { value, children: /* @__PURE__ */ jsxRuntime.jsx(FormElementDescriptorsContext.HvFormElementDescriptorsContextProvider, { value: descriptors, children }) }) }) });
};
HvFormElement.formElementType = "formelement";
exports.formElementClasses = FormElement_styles.staticClasses;
exports.HvFormElement = HvFormElement;
