// @hitachivantara/uikit-react-core/dist/esm/components/TimePicker/Unit/Unit.js
import { jsxs, jsx } from "@emotion/react/jsx-runtime";
import { DropUpXS, DropDownXS } from "@hitachivantara/uikit-react-icons";
import { theme } from "@hitachivantara/uikit-styles";
import { useClasses } from "./Unit.styles.js";
import { HvToggleButton } from "../../ToggleButton/ToggleButton.js";
import { HvInput } from "../../Input/Input.js";
const Unit = ({
  id,
  state,
  segment,
  placeholder: placeholderProp,
  onChange,
  onAdd,
  onSub
}) => {
  const {
    classes
  } = useClasses();
  const {
    type,
    text
  } = segment;
  const placeholder = placeholderProp ?? segment.placeholder;
  return /* @__PURE__ */ jsxs("div", { className: classes.root, children: [
    type !== "literal" && /* @__PURE__ */ jsx(DropUpXS, { onClick: onAdd }),
    type === "literal" && /* @__PURE__ */ jsx("div", { className: classes.separator, children: text }),
    type === "dayPeriod" && /* @__PURE__ */ jsx(HvToggleButton, { className: classes.periodToggle, onClick: onAdd, children: text }),
    ["hour", "minute", "second"].includes(type) && /* @__PURE__ */ jsx(HvInput, { id, disableClear: true, style: {
      ...theme.typography.title3
    }, classes: {
      input: classes.input,
      root: classes.inputContainer,
      inputBorderContainer: classes.inputBorderContainer,
      inputRoot: classes.inputRoot
    }, onKeyDown: (event) => {
      if (event.key === "Enter") {
        event.preventDefault();
        event.stopPropagation();
      }
    }, required: true, status: state.validationState, value: text.padStart(2, "0"), onChange, placeholder, inputProps: {
      autoComplete: "off",
      type: "number"
    } }),
    type !== "literal" && /* @__PURE__ */ jsx(DropDownXS, { onClick: onSub })
  ] });
};
export {
  Unit
};
