// @hitachivantara/uikit-react-core/dist/cjs/components/Card/Header/Header.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const MuiCardHeader = require("@mui/material/CardHeader");
const useTheme = require("../../../hooks/useTheme.cjs");
const Header_styles = require("./Header.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const MuiCardHeader__default = /* @__PURE__ */ _interopDefault(MuiCardHeader);
const HvCardHeader = ({
  classes: classesProp,
  className,
  title,
  subheader,
  icon,
  onClick,
  ...others
}) => {
  const {
    activeTheme
  } = useTheme.useTheme();
  const {
    classes,
    css,
    cx
  } = Header_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(MuiCardHeader__default.default, { title, subheader, action: icon, onClick, className: cx(classes.root, className), classes: {
    title: cx(css({
      ...activeTheme == null ? void 0 : activeTheme.typography[activeTheme == null ? void 0 : activeTheme.card.titleVariant]
    }), {
      [classes.titleShort]: icon,
      [classes.title]: !icon
    }),
    subheader: cx(css({
      ...activeTheme == null ? void 0 : activeTheme.typography[activeTheme == null ? void 0 : activeTheme.card.subheaderVariant],
      color: activeTheme == null ? void 0 : activeTheme.card.subheaderColor
    }), classes.subheader),
    action: classes.action,
    content: classes.content
  }, ...others });
};
exports.cardHeaderClasses = Header_styles.staticClasses;
exports.HvCardHeader = HvCardHeader;
