// @hitachivantara/uikit-react-core/dist/cjs/components/BaseInput/BaseInput.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvBaseInput", {
  root: {
    display: "inline-block",
    width: "100%",
    position: "relative",
    "&:hover $inputBorderContainer": {
      backgroundColor: uikitStyles.theme.baseInput.hoverColor
    },
    "&:focus-within $inputBorderContainer": {
      backgroundColor: uikitStyles.theme.baseInput.hoverColor
    }
  },
  disabled: {
    "& $inputBorderContainer": {
      backgroundColor: uikitStyles.theme.colors.atmo4
    },
    "&:hover $inputBorderContainer": {
      backgroundColor: uikitStyles.theme.colors.atmo4
    },
    "&& $input": {
      color: uikitStyles.theme.baseInput.disabledTextColor,
      WebkitTextFillColor: uikitStyles.theme.baseInput.disabledTextColor
    },
    "& $inputRootMultiline": {
      "& $input": {
        backgroundColor: uikitStyles.theme.baseInput.disabledBackgroundColor,
        border: `1px solid ${uikitStyles.theme.baseInput.multilineDisabledBorderColor}`
      }
    },
    "&:hover $inputRootMultiline": {
      "& $input": {
        backgroundColor: uikitStyles.theme.baseInput.disabledBackgroundColor,
        border: `1px solid ${uikitStyles.theme.baseInput.multilineDisabledBorderColor}`
      }
    }
  },
  invalid: {
    "&:not(.disabled)": {
      "& $inputBorderContainer": {
        backgroundColor: uikitStyles.theme.colors.negative
      },
      "&:hover $inputBorderContainer": {
        backgroundColor: uikitStyles.theme.colors.negative
      },
      "& $inputRootMultiline": {
        "& $input": {
          border: `1px solid ${uikitStyles.theme.colors.negative}`
        }
      },
      "&:hover $inputRootMultiline": {
        "& $input": {
          border: `1px solid ${uikitStyles.theme.colors.negative}`
        }
      },
      "&:focus-within $inputRootMultiline": {
        "& $input": {
          border: `1px solid ${uikitStyles.theme.colors.negative}`
        }
      }
    }
  },
  resizable: {
    width: "auto"
  },
  readOnly: {
    "& $inputBorderContainer": {
      backgroundColor: "transparent"
    },
    "&:hover $inputBorderContainer": {
      backgroundColor: "transparent"
    },
    "&:focus-within $inputBorderContainer": {
      backgroundColor: "transparent"
    },
    "& $inputRootMultiline": {
      "& $input": {
        border: `1px solid ${uikitStyles.theme.baseInput.readOnlyBorderColor}`,
        backgroundColor: uikitStyles.theme.baseInput.readOnlyBackgroundColor
      }
    },
    "&:hover $inputRootMultiline": {
      "& $input": {
        border: `1px solid ${uikitStyles.theme.baseInput.readOnlyBorderColor}`,
        backgroundColor: uikitStyles.theme.baseInput.readOnlyBackgroundColor
      }
    },
    "&:focus-within $inputRootMultiline": {
      "& $input": {
        border: `1px solid ${uikitStyles.theme.baseInput.readOnlyBorderColor}`,
        backgroundColor: uikitStyles.theme.baseInput.readOnlyBackgroundColor
      }
    }
  },
  inputBorderContainer: {
    position: "absolute",
    width: "calc(100% - 4px)",
    height: uikitStyles.theme.baseInput.underlineHeight,
    top: "31px",
    left: "2px",
    backgroundColor: uikitStyles.theme.colors.atmo4
  },
  inputRootInvalid: {
    borderColor: uikitStyles.theme.colors.negative
  },
  inputRootReadOnly: {
    borderColor: uikitStyles.theme.baseInput.readOnlyBorderColor,
    backgroundColor: uikitStyles.theme.baseInput.readOnlyBackgroundColor
  },
  inputRoot: {
    margin: 0,
    width: "100%",
    borderRadius: uikitStyles.theme.radii.base,
    height: "32px",
    border: `1px solid ${uikitStyles.theme.baseInput.borderColor}`,
    boxSizing: "border-box",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    fontFamily: uikitStyles.theme.fontFamily.body,
    "&:hover:not($inputRootDisabled):not($inputRootInvalid):not($inputRootReadOnly)": {
      borderColor: uikitStyles.theme.baseInput.hoverColor
    },
    "&:hover:not($inputRootDisabled)::before": {
      borderBottom: "none"
    },
    "&::before": {
      borderBottom: "none"
    },
    "&::after": {
      borderBottom: "none"
    }
  },
  inputRootFocused: {
    background: uikitStyles.theme.colors.atmo1,
    ...focusUtils.outlineStyles,
    "&:hover": {
      background: uikitStyles.theme.colors.atmo1
    },
    "& $inputRootReadOnly": {
      backgroundColor: uikitStyles.theme.baseInput.readOnlyBackgroundColor
    }
  },
  inputRootDisabled: {
    background: uikitStyles.theme.baseInput.disabledBackgroundColor,
    borderColor: uikitStyles.theme.baseInput.disabledBorderColor,
    cursor: "not-allowed",
    "&:hover": {
      background: uikitStyles.theme.baseInput.disabledBackgroundColor,
      cursor: "not-allowed"
    },
    "&&::before": {
      borderBottomStyle: "none"
    }
  },
  inputRootMultiline: {
    padding: 0,
    backgroundColor: "transparent",
    overflow: "auto",
    border: "none",
    height: "auto",
    "& $input": {
      border: `1px solid ${uikitStyles.theme.baseInput.multilineBorderColor}`,
      borderRadius: uikitStyles.theme.radii.base,
      backgroundColor: uikitStyles.theme.colors.atmo1,
      height: "auto",
      minHeight: "21px",
      padding: "5px 10px",
      overflow: "auto",
      marginLeft: "0px",
      marginRight: "0px",
      "&:hover": {
        border: `1px solid ${uikitStyles.theme.baseInput.hoverColor}`
      }
    }
  },
  input: {
    height: "19px",
    marginLeft: uikitStyles.theme.space.xs,
    marginRight: uikitStyles.theme.space.xs,
    padding: "6px 0 5px",
    overflow: "hidden",
    textOverflow: "ellipsis",
    outline: "none",
    width: "initial",
    flexGrow: 1,
    ...uikitStyles.theme.typography.body,
    "&::placeholder": {
      opacity: 1,
      color: uikitStyles.theme.baseInput.placeholderColor
    },
    "&::-ms-clear": {
      display: "none"
    }
  },
  inputDisabled: {},
  inputReadOnly: {
    color: uikitStyles.theme.baseInput.readOnlyTextColor
  },
  inputResizable: {
    resize: "both",
    width: "100%"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
