// @hitachivantara/uikit-react-core/dist/esm/components/AppSwitcher/AppSwitcher.js
import { jsx, jsxs } from "@emotion/react/jsx-runtime";
import { useMemo } from "react";
import { useDefaultProps } from "../../hooks/useDefaultProps.js";
import { useClasses } from "./AppSwitcher.styles.js";
import { staticClasses } from "./AppSwitcher.styles.js";
import TitleWithTooltip from "./TitleWithTooltip.js";
import { HvAppSwitcherAction } from "./Action/Action.js";
import { HvTypography } from "../Typography/Typography.js";
import { HvListContainer } from "../ListContainer/ListContainer.js";
const HvAppSwitcher = (props) => {
  const {
    id,
    className,
    classes: classesProp,
    layout = "single",
    title,
    applications,
    onActionClickedCallback = () => {
    },
    isActionSelectedCallback = () => false,
    header,
    footer,
    isOpen
  } = useDefaultProps("HvAppSwitcher", props);
  const {
    classes,
    cx
  } = useClasses(classesProp);
  const panelActions = useMemo(() => applications && (applications == null ? void 0 : applications.map((application) => {
    if (application.name) {
      return /* @__PURE__ */ jsx(HvAppSwitcherAction, { application, onClickCallback: onActionClickedCallback, isSelectedCallback: isActionSelectedCallback, classes: {
        root: classes.item,
        selected: classes.itemSelected,
        disabled: classes.itemDisabled,
        typography: classes.itemTrigger,
        icon: classes.itemIcon,
        title: classes.itemTitle,
        iconInfo: classes.itemInfoIcon
      } }, application.id || `${application.name}_${application.url}`);
    }
    return void 0;
  })), [applications, classes, isActionSelectedCallback, onActionClickedCallback]);
  return /* @__PURE__ */ jsxs("div", { id, className: cx(classes.root, classes[layout], {
    [classes.open]: !!isOpen,
    [classes.closed]: isOpen === false
  }, className), children: [
    header && /* @__PURE__ */ jsx(HvTypography, { component: "div", variant: "label", className: classes.title, children: header }) || title && /* @__PURE__ */ jsx(TitleWithTooltip, { className: classes.title, title }),
    /* @__PURE__ */ jsx(HvListContainer, { disableGutters: true, className: classes.actionsContainer, children: panelActions }),
    footer && /* @__PURE__ */ jsx(HvTypography, { component: "div", variant: "label", className: classes.footerContainer, children: footer })
  ] });
};
export {
  HvAppSwitcher,
  staticClasses as appSwitcherClasses
};
