// @hitachivantara/uikit-react-core/dist/cjs/components/Snackbar/SnackbarContent/SnackbarContent.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const ActionsGeneric_styles = require("../../ActionsGeneric/ActionsGeneric.styles.cjs");
require("@emotion/react/jsx-runtime");
require("react");
require("@hitachivantara/uikit-react-icons");
const {
  useClasses,
  staticClasses
} = classes.createClasses("HvSnackbar-Content", {
  root: {
    width: "310px",
    minHeight: "52px",
    maxHeight: "92px",
    padding: uikitStyles.theme.space.xs,
    boxShadow: "none"
  },
  success: {
    backgroundColor: uikitStyles.theme.colors.positive_20
  },
  error: {
    backgroundColor: uikitStyles.theme.colors.negative_20
  },
  default: {
    backgroundColor: uikitStyles.theme.colors.neutral_20
  },
  warning: {
    backgroundColor: uikitStyles.theme.colors.warning_20
  },
  message: {
    padding: 0,
    width: "100%"
  },
  messageSpan: {
    display: "flex",
    alignItems: "center",
    minHeight: "32px"
  },
  messageText: {
    paddingLeft: uikitStyles.theme.space.xs,
    color: uikitStyles.theme.colors.base_dark,
    fontFamily: uikitStyles.theme.fontFamily.body,
    maxHeight: "72px",
    wordBreak: "break-word",
    textWrap: "balance",
    overflow: "hidden"
  },
  action: {
    textAlign: "right",
    paddingLeft: uikitStyles.theme.space.xs,
    marginLeft: uikitStyles.theme.snackbar.actionMarginLeft,
    [`& .${ActionsGeneric_styles.staticClasses.button}`]: {
      borderColor: uikitStyles.theme.colors.base_dark,
      color: uikitStyles.theme.colors.base_dark,
      "&:hover": {
        borderColor: uikitStyles.theme.colors.base_dark
      }
    }
  },
  iconVariant: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
