// @hitachivantara/uikit-react-core/dist/cjs/components/FilterGroup/FilterGroup.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const useLabels = require("../../hooks/useLabels.cjs");
const useUniqueId = require("../../hooks/useUniqueId.cjs");
const useControlled = require("../../hooks/useControlled.cjs");
const setId = require("../../utils/setId.cjs");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const FilterGroup_styles = require("./FilterGroup.styles.cjs");
const FilterGroupContext = require("./FilterGroupContext.cjs");
const FilterContent = require("./FilterContent/FilterContent.cjs");
const FormElement = require("../Forms/FormElement/FormElement.cjs");
const Label = require("../Forms/Label/Label.cjs");
const InfoMessage = require("../Forms/InfoMessage/InfoMessage.cjs");
const WarningText = require("../Forms/WarningText/WarningText.cjs");
const DEFAULT_LABELS = {
  applyLabel: "Apply",
  cancelLabel: "Cancel",
  clearLabel: "Clear Filters",
  placeholder: "Filters",
  searchBoxPlaceholder: "Search",
  selectAll: "All",
  multiSelectionConjunction: "/"
};
const HvFilterGroup = (props) => {
  const {
    className,
    id,
    name,
    required = false,
    disabled = false,
    label,
    "aria-label": ariaLabel,
    "aria-labelledby": ariaLabelledBy,
    description,
    "aria-describedby": ariaDescribedBy,
    onChange,
    onCancel,
    onClear,
    status,
    statusMessage,
    labels: labelsProp,
    defaultValue,
    value,
    filters,
    horizontalPlacement = "right",
    disablePortal = true,
    escapeWithReference = true,
    height = 350,
    filterContentProps,
    classes: classesProp,
    ...others
  } = useDefaultProps.useDefaultProps("HvFilterGroup", props);
  const {
    classes,
    cx
  } = FilterGroup_styles.useClasses(classesProp);
  const [validationMessage] = useControlled.useControlled(statusMessage, "Required");
  const elementId = useUniqueId.useUniqueId(id, "hvfiltergroup");
  const labels = useLabels.useLabels(DEFAULT_LABELS, labelsProp);
  const hasLabel = label != null;
  const hasDescription = description != null;
  const canShowError = status !== void 0 || required;
  return /* @__PURE__ */ jsxRuntime.jsxs(FormElement.HvFormElement, { id, name, value, status, disabled, required, className: cx(classes.root, className), ...others, children: [
    (hasLabel || hasDescription) && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: classes.labelContainer, children: [
      hasLabel && /* @__PURE__ */ jsxRuntime.jsx(Label.HvLabel, { id: setId.setId(elementId, "label"), htmlFor: setId.setId(elementId, "input"), label, className: classes.label }),
      hasDescription && /* @__PURE__ */ jsxRuntime.jsx(InfoMessage.HvInfoMessage, { id: setId.setId(elementId, "description"), className: classes.description, children: description })
    ] }),
    /* @__PURE__ */ jsxRuntime.jsxs(FilterGroupContext.HvFilterGroupProvider, { defaultValue, value, filters, children: [
      /* @__PURE__ */ jsxRuntime.jsx(FilterContent.HvFilterGroupContent, { id: elementId, disabled, disablePortal, variableWidth: true, placement: horizontalPlacement, escapeWithReference, "aria-label": ariaLabel, "aria-labelledby": ariaLabelledBy, status, onChange, onCancel, onClear, labels, height, ...filterContentProps }),
      canShowError && /* @__PURE__ */ jsxRuntime.jsx(WarningText.HvWarningText, { id: setId.setId(elementId, "error"), disableBorder: true, className: classes.error, children: validationMessage })
    ] })
  ] });
};
exports.filterGroupClasses = FilterGroup_styles.staticClasses;
exports.HvFilterGroup = HvFilterGroup;
