// @hitachivantara/uikit-react-core/dist/cjs/components/OverflowTooltip/OverflowTooltip.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const React = require("react");
const reactResizeDetector = require("react-resize-detector");
const OverflowTooltip_styles = require("./OverflowTooltip.styles.cjs");
const Tooltip = require("../Tooltip/Tooltip.cjs");
const Typography = require("../Typography/Typography.cjs");
const isParagraph = (children = "") => /\s/.test(children);
const HvOverflowTooltip = (props) => {
  var _a;
  const {
    id,
    classes: classesProp,
    className,
    data,
    open,
    paragraphOverflow,
    placement = "top-start",
    tooltipsProps
  } = useDefaultProps.useDefaultProps("HvOverflowTooltip", props);
  const {
    classes,
    cx
  } = OverflowTooltip_styles.useClasses(classesProp);
  const {
    width = 0,
    ref
  } = reactResizeDetector.useResizeDetector({
    refreshMode: "debounce",
    refreshOptions: {
      trailing: true
    },
    handleHeight: false
  });
  const scrollWidth = ((_a = ref.current) == null ? void 0 : _a.scrollWidth) || 0;
  const isOverflowing = scrollWidth - width >= 1;
  const isParag = React.useMemo(() => paragraphOverflow && isParagraph(data == null ? void 0 : data.toString()), [data, paragraphOverflow]);
  const content = React.useMemo(() => /* @__PURE__ */ jsxRuntime.jsx("div", { ref, className: cx({
    [classes.tooltipAnchor]: !isParag,
    [classes.tooltipAnchorParagraph]: isParag
  }, className), children: data }), [className, classes.tooltipAnchor, classes.tooltipAnchorParagraph, cx, data, isParag, ref]);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tooltip.HvTooltip,
    {
      id,
      disableHoverListener: !isOverflowing,
      open,
      placement,
      title: /* @__PURE__ */ jsxRuntime.jsx(Typography.HvTypography, { className: classes.tooltipData, variant: "body", children: data }),
      "aria-label": null,
      "aria-labelledby": null,
      ...tooltipsProps,
      children: content
    }
  );
};
exports.overflowTooltipClasses = OverflowTooltip_styles.staticClasses;
exports.HvOverflowTooltip = HvOverflowTooltip;
