// @hitachivantara/uikit-react-core/dist/cjs/components/ScrollTo/Vertical/ScrollToVertical.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const name = "HvScrollToVertical";
const {
  staticClasses,
  useClasses
} = classes.createClasses(name, {
  root: {
    display: "flex",
    width: "32px",
    padding: "0",
    margin: 0,
    listStyleType: "none",
    flexWrap: "wrap",
    flexDirection: "column",
    backdropFilter: `blur(${uikitStyles.theme.scrollTo.backgroundColorBlur})`
  },
  positionAbsolute: {
    width: "32px",
    position: "absolute",
    zIndex: `calc(${uikitStyles.theme.zIndices.banner} - 2)`,
    right: "0"
  },
  positionFixed: {
    width: "32px",
    position: "fixed",
    zIndex: `calc(${uikitStyles.theme.zIndices.banner} - 2)`,
    right: "0"
  }
});
const calculateOffset = (quantityOfOptions) => {
  const itemSize = 32;
  const halfOptions = Math.round(quantityOfOptions * 0.5);
  return halfOptions * itemSize;
};
exports.calculateOffset = calculateOffset;
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
