// @hitachivantara/uikit-react-core/dist/esm/components/BaseCheckBox/BaseCheckBox.js
import { jsx } from "@emotion/react/jsx-runtime";
import { useState, useCallback } from "react";
import { Checkbox, CheckboxPartial, CheckboxCheck } from "@hitachivantara/uikit-react-icons";
import { useDefaultProps } from "../../hooks/useDefaultProps.js";
import MuiCheckbox from "@mui/material/Checkbox";
import { useClasses } from "./BaseCheckBox.styles.js";
import { staticClasses } from "./BaseCheckBox.styles.js";
const getSelectorIcons = (options, classes) => {
  const {
    disabled,
    semantic
  } = options;
  const color = disabled && ["atmo3", "secondary_60"] || semantic && ["base_light", "base_dark"] || void 0;
  const checkedColor = disabled && ["atmo3", "secondary_60"] || semantic && ["base_dark", "base_light"] || void 0;
  return {
    checkbox: /* @__PURE__ */ jsx(Checkbox, { color, className: classes.icon }),
    checkboxPartial: /* @__PURE__ */ jsx(CheckboxPartial, { color, className: classes.icon }),
    checkboxChecked: /* @__PURE__ */ jsx(CheckboxCheck, { color: checkedColor, className: classes.icon })
  };
};
const HvBaseCheckBox = (props) => {
  const {
    id,
    classes: classesProp,
    className,
    name,
    inputProps,
    onChange,
    onFocusVisible,
    onBlur,
    checked,
    indeterminate,
    defaultChecked,
    value = "on",
    required = false,
    readOnly = false,
    disabled = false,
    semantic = false,
    ...others
  } = useDefaultProps("HvBaseCheckBox", props);
  const {
    classes,
    cx
  } = useClasses(classesProp);
  const [focusVisible, setFocusVisible] = useState(false);
  const icons = getSelectorIcons({
    disabled,
    semantic
  }, classes);
  const onChangeCallback = useCallback((event) => {
    if (readOnly) {
      return;
    }
    onChange == null ? void 0 : onChange(event, event.target.checked, value);
  }, [onChange, readOnly, value]);
  const onBlurCallback = useCallback((event) => {
    setFocusVisible(false);
    onBlur == null ? void 0 : onBlur(event);
  }, [onBlur]);
  const onFocusVisibleCallback = useCallback((event) => {
    setFocusVisible(true);
    onFocusVisible == null ? void 0 : onFocusVisible(event);
  }, [onFocusVisible]);
  return /* @__PURE__ */ jsx(MuiCheckbox, { id, name, value, className: cx(classes.root, {
    [classes.disabled]: disabled,
    [classes.focusVisible]: focusVisible
  }, className), icon: icons.checkbox, indeterminateIcon: icons.checkboxPartial, checkedIcon: icons.checkboxChecked, disabled, required, readOnly, checked, defaultChecked, indeterminate, inputProps, onChange: onChangeCallback, onFocusVisible: onFocusVisibleCallback, onBlur: onBlurCallback, color: "default", disableRipple: true, ...others });
};
export {
  HvBaseCheckBox,
  staticClasses as baseCheckBoxClasses
};
