// @hitachivantara/uikit-react-core/dist/cjs/components/Grid/Grid.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const material = require("@mui/material");
const isString = require("lodash/isString");
const React = require("react");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const Grid_styles = require("./Grid.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const isString__default = /* @__PURE__ */ _interopDefault(isString);
const BREAKPOINT_GUTTERS = {
  xs: 2,
  sm: 2,
  md: 4,
  lg: 4,
  xl: 4
};
const BREAKPOINT_COLUMNS = {
  xs: 4,
  sm: 8,
  md: 12,
  lg: 12,
  xl: 12
};
function getGridSpacing(spacing) {
  let gridSpacing;
  if (isString__default.default(spacing)) {
    if (spacing === "auto") {
      gridSpacing = BREAKPOINT_GUTTERS;
    } else {
      gridSpacing = BREAKPOINT_GUTTERS[spacing];
    }
  } else if (typeof spacing === "object") {
    gridSpacing = Object.keys(spacing).reduce((acc, breakpoint) => ({
      ...acc,
      [breakpoint]: BREAKPOINT_GUTTERS[spacing[breakpoint]] ?? spacing[breakpoint]
    }), {});
  } else if (spacing === 0) {
    gridSpacing = {
      xs: 0
    };
  } else {
    gridSpacing = spacing;
  }
  return gridSpacing;
}
function getNumberOfColumns(columns) {
  let numberOfColumns;
  if (columns === "auto") {
    numberOfColumns = BREAKPOINT_COLUMNS;
  } else {
    numberOfColumns = columns;
  }
  return numberOfColumns;
}
const HvGrid = React.forwardRef((props, ref) => {
  const {
    container,
    spacing = "auto",
    rowSpacing,
    columnSpacing,
    columns,
    classes: classesProp,
    ...others
  } = useDefaultProps.useDefaultProps("HvGrid", props);
  const {
    classes
  } = Grid_styles.useClasses(classesProp);
  const containerProps = {};
  if (container) {
    containerProps.container = true;
    if (spacing != null) {
      containerProps.spacing = getGridSpacing(spacing);
    }
    if (rowSpacing != null) {
      containerProps.rowSpacing = getGridSpacing(rowSpacing);
    }
    if (columnSpacing != null) {
      containerProps.columnSpacing = getGridSpacing(columnSpacing);
    }
    if (columns != null) {
      containerProps.columns = getNumberOfColumns(columns);
    }
  }
  return /* @__PURE__ */ jsxRuntime.jsx(material.Grid, { ref, classes, ...containerProps, ...others });
});
exports.gridClasses = Grid_styles.staticClasses;
exports.HvGrid = HvGrid;
