// @hitachivantara/uikit-react-core/dist/cjs/components/DropDownMenu/DropDownMenu.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvDropDownMenu", {
  /** Styles applied to the container. */
  container: {
    width: 32
  },
  /** Styles applied to the BaseDropdown container. */
  baseContainer: {},
  /** Styles applied to the root. */
  root: {
    display: "inline-block",
    width: "auto",
    "&:focus-visible $icon": {
      ...focusUtils.outlineStyles
    }
  },
  /** Styles applied to the icon. */
  icon: {
    position: "relative",
    width: 32,
    height: 32,
    boxSizing: "border-box",
    padding: 0,
    borderRadius: uikitStyles.theme.dropDownMenu.borderRadius,
    border: uikitStyles.theme.dropDownMenu.borderClosed
  },
  /** Styles applied to the icon when selected. */
  iconSelected: {
    backgroundColor: uikitStyles.theme.colors.atmo1,
    boxShadow: uikitStyles.theme.colors.shadow,
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.atmo1
    },
    "& svg .color0": {
      fill: uikitStyles.theme.colors.secondary
    },
    borderRadius: `${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} 0px 0px`,
    border: uikitStyles.theme.dropDownMenu.borderOpened
  },
  /** Styles applied to the list root. */
  menuListRoot: {},
  /** Styles applied to the list. */
  menuList: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
