// @hitachivantara/uikit-react-core/dist/cjs/components/FileUploader/DropZone/DropZone.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const classes = require("../../../utils/classes.cjs");
const focusUtils = require("../../../utils/focusUtils.cjs");
const uikitStyles = require("@hitachivantara/uikit-styles");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvDropZone", {
  dropZoneContainer: {
    position: "relative",
    width: "100%",
    display: "flex",
    border: `1px ${uikitStyles.theme.fileUploader.dropZone.borderType} ${uikitStyles.theme.fileUploader.dropZone.borderColor}`,
    cursor: "pointer",
    background: uikitStyles.theme.fileUploader.dropZone.backgroundColor,
    borderRadius: uikitStyles.theme.fileUploader.dropZone.borderRadius,
    "&:hover": {
      background: `${uikitStyles.theme.colors.atmo1}`,
      border: `1px ${uikitStyles.theme.fileUploader.dropZone.borderType} ${uikitStyles.theme.colors.secondary}`
    },
    "&:focus-within": {
      background: `${uikitStyles.theme.colors.atmo1}`,
      border: `1px ${uikitStyles.theme.fileUploader.dropZone.borderType} ${uikitStyles.theme.colors.secondary}`,
      ...focusUtils.outlineStyles
    }
  },
  dropZoneLabelsGroup: {
    display: "flex",
    justifyContent: "start",
    "& label:nth-of-type(1)": {},
    "& p:nth-of-type(2)": {
      marginLeft: "auto"
    }
  },
  dragAction: {
    background: `${uikitStyles.theme.colors.atmo1}`,
    border: `1px ${uikitStyles.theme.fileUploader.dropZone.borderType} ${uikitStyles.theme.fileUploader.dropZone.borderColorDrag}`
  },
  dropZoneContainerDisabled: {
    background: `${uikitStyles.theme.colors.atmo3}`,
    border: `1px ${uikitStyles.theme.fileUploader.dropZone.borderType} ${uikitStyles.theme.fileUploader.dropZone.borderColorDisabled}`,
    cursor: "not-allowed",
    "&:hover": {
      background: `${uikitStyles.theme.colors.atmo3}`,
      border: `1px ${uikitStyles.theme.fileUploader.dropZone.borderType} ${uikitStyles.theme.fileUploader.dropZone.borderColorDisabled}`
    },
    "& $dragText": {
      color: uikitStyles.theme.colors.secondary_60
    },
    "& $selectFilesText": {
      color: uikitStyles.theme.colors.secondary_60
    }
  },
  inputArea: {
    opacity: 0,
    width: "100%",
    position: "absolute",
    height: "100%",
    cursor: "pointer",
    "&:disabled": {
      cursor: "not-allowed"
    }
  },
  dropArea: {
    display: "flex",
    margin: `${uikitStyles.theme.space.md} auto`,
    minHeight: 48
  },
  dropZoneAreaLabels: {
    display: "flex",
    maxWidth: 120,
    margin: "auto"
  },
  dropZoneAreaIcon: {
    margin: "auto",
    marginRight: uikitStyles.theme.space.xs
  },
  dropZoneLabel: {
    paddingBottom: 6
  },
  dragText: {
    ...uikitStyles.theme.typography.body
  },
  selectFilesText: {
    ...uikitStyles.theme.typography.label
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
