// @hitachivantara/uikit-react-core/dist/cjs/components/GlobalActions/GlobalActions.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvGlobalActions", {
  root: {
    position: "relative"
  },
  positionSticky: {
    width: "100%",
    position: "sticky"
  },
  positionFixed: {
    width: "100%",
    position: "fixed"
  },
  global: {
    zIndex: `calc(${uikitStyles.theme.zIndices.banner} - 2)`,
    top: 0,
    left: 0,
    padding: uikitStyles.theme.spacing(1, 0),
    backdropFilter: "blur(1px)",
    "&:before": {
      content: "''",
      display: "flex",
      position: "absolute",
      inset: 0,
      background: uikitStyles.theme.colors.atmo2,
      opacity: "75%"
    }
  },
  wrapper: {
    padding: uikitStyles.theme.space.sm,
    display: "flex",
    justifyContent: "flex-start",
    alignItems: "center",
    border: uikitStyles.theme.globalActions.border,
    borderRadius: uikitStyles.theme.globalActions.borderRadius
  },
  globalWrapperComplement: {
    position: "relative",
    top: 0,
    left: 0,
    background: uikitStyles.theme.colors.atmo1,
    width: "100%"
  },
  globalSectionArea: {
    backgroundColor: uikitStyles.theme.globalActions.sectionBackgroundColor,
    borderTop: `1px solid ${uikitStyles.theme.colors.atmo4}`,
    paddingLeft: uikitStyles.theme.globalActions.sectionPaddingLeft
  },
  backButton: {
    marginRight: uikitStyles.theme.space.xs
  },
  name: {
    flexGrow: 1
  },
  actions: {
    display: "flex",
    alignItems: "center",
    marginLeft: "auto",
    gap: uikitStyles.theme.space.xs
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
