// @hitachivantara/uikit-react-core/dist/cjs/components/Focus/Focus.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const focusUtils = require("../../utils/focusUtils.cjs");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvFocus", {
  root: {},
  selected: {},
  disabled: {},
  focusDisabled: {
    outline: "none",
    "& *:focus": {
      outline: "none"
    },
    "& *": {
      outline: "none !important"
    }
  },
  focused: {
    ...focusUtils.outlineStyles,
    "@media (-webkit-min-device-pixel-ratio:0)": {
      ...focusUtils.outlineStyles
    }
  },
  externalReference: {
    position: "relative"
  },
  falseFocus: {
    width: "98%",
    height: "98%",
    position: "absolute",
    zIndex: "1",
    ...focusUtils.outlineStyles,
    "@media (-webkit-min-device-pixel-ratio:0)": {
      ...focusUtils.outlineStyles
    },
    top: 0,
    left: "0.5%",
    backgroundColor: "transparent",
    pointerEvents: "none"
  },
  focus: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
