// @hitachivantara/uikit-react-core/dist/cjs/components/Calendar/CalendarHeader/CalendarHeader.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const classes = require("../../../utils/classes.cjs");
const focusUtils = require("../../../utils/focusUtils.cjs");
const uikitStyles = require("@hitachivantara/uikit-styles");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvCalendarHeader", {
  root: {
    marginTop: uikitStyles.theme.spacing("xs"),
    backgroundColor: uikitStyles.theme.colors.atmo1,
    borderBottom: uikitStyles.theme.calendar.headerInputBorderBottom,
    borderTop: uikitStyles.theme.calendar.headerInputBorderTop,
    borderLeft: uikitStyles.theme.calendar.headerInputBorderLeft,
    borderRight: uikitStyles.theme.calendar.headerInputBorderRight,
    borderRadius: uikitStyles.theme.calendar.borderRadius,
    "&:hover, &:focus": {
      borderBottom: `1px solid ${uikitStyles.theme.colors.secondary}`
    }
  },
  invalid: {},
  headerDayOfWeek: {
    color: uikitStyles.theme.calendar.headerInputFontColor,
    paddingLeft: uikitStyles.theme.spacing("xs")
  },
  headerDate: {},
  input: {
    border: "none",
    backgroundColor: "transparent",
    padding: `5px ${uikitStyles.theme.spacing("xs")}`,
    fontFamily: uikitStyles.theme.fontFamily.body,
    color: uikitStyles.theme.calendar.headerInputFontColor,
    fontSize: uikitStyles.theme.calendar.headerInputFontSize,
    letterSpacing: uikitStyles.theme.calendar.headerInputFontLetterSpacing,
    lineHeight: uikitStyles.theme.calendar.headerInputFontLineHeight,
    fontWeight: uikitStyles.theme.calendar.headerInputFontWeight,
    width: "100%",
    "&::placeholder": {
      color: uikitStyles.theme.colors.secondary_60
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    }
  },
  inputBorderContainer: {
    backgroundColor: uikitStyles.theme.colors.negative,
    height: 1,
    marginTop: -1
  },
  invalidMessageStyling: {
    display: "flex",
    alignItems: "center",
    color: uikitStyles.theme.colors.negative
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
