// @hitachivantara/app-shell-vite-plugin/dist/vite-generate-base-plugin.js
import { getPublicPath } from "./config-utils.js";
/**
 * Add the <BASE> tag at index.html file. Tag value is obtained by the main app ({@link #getMainApp}) baseUrl ({@link #getPublicPath})‘
 * @param appShellConfig The app shell configuration.
 * @param generateEmptyShell Flag that controls if an empty app shell application is created with tags instead of final values
 */
export default function generateBaseTag(appShellConfig, generateEmptyShell) {
    const publicPath = getPublicPath(appShellConfig);
    return {
        name: "vite-plugin-generate-base",
        enforce: "pre",
        transformIndexHtml: {
            enforce: "post",
            transform: (html) => ({
                html: generateEmptyShell || appShellConfig.name != null
                    ? html.replace(/<title>(.*?)<\/title>/, `<title>${generateEmptyShell
                        ? "%%APPSHELL_TITLE%%"
                        : appShellConfig.name}</title>`)
                    : html,
                tags: [
                    {
                        tag: "base",
                        attrs: {
                            href: generateEmptyShell ? "%%APPSHELL_BASE%%" : publicPath
                        },
                        injectTo: "head-prepend"
                    },
                    {
                        tag: "link",
                        injectTo: "head-prepend",
                        attrs: {
                            rel: "preload",
                            as: "image",
                            href: "./icons/icons.svg"
                        }
                    }
                ]
            })
        }
    };
}
