// @hitachivantara/uikit-react-core/dist/cjs/components/QueryBuilder/Rule/Rule.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const QueryBuilder_styles = require("../QueryBuilder.styles.cjs");
const {
  useClasses,
  staticClasses
} = classes.createClasses("HvQueryBuilder-Rule", {
  root: {
    position: "relative",
    marginTop: uikitStyles.theme.space.xs,
    minHeight: 94,
    "&>div:not(:last-child)": {
      marginRight: uikitStyles.theme.space.md
    },
    // hide required * as all fields are required
    "& label>span[aria-hidden]": {
      visibility: "hidden"
    },
    "&::before": {
      content: '""',
      position: "absolute",
      zIndex: 2,
      width: uikitStyles.theme.queryBuilder.ruleConnectorHorizontalSize,
      height: uikitStyles.theme.queryBuilder.ruleConnectorHeight,
      borderBottom: uikitStyles.theme.queryBuilder.border,
      borderLeft: uikitStyles.theme.queryBuilder.border,
      top: 0,
      left: uikitStyles.theme.queryBuilder.ruleLeftConnectorPosition
    },
    [`:not(.${QueryBuilder_styles.staticClasses.topRulesContainer})>&:last-child::after`]: {
      content: '""',
      position: "absolute",
      zIndex: 1,
      width: uikitStyles.theme.queryBuilder.ruleConnectorHorizontalSize,
      height: "100%",
      borderLeft: uikitStyles.theme.queryBuilder.border,
      top: 0,
      left: uikitStyles.theme.queryBuilder.ruleLeftConnectorPosition
    }
  },
  actionsContainer: {
    marginLeft: "auto",
    marginTop: uikitStyles.theme.queryBuilder.actionsContainerMarginTop,
    "&>:not(:last-child)": {
      marginRight: uikitStyles.theme.space.xs
    }
  },
  isMdDown: {
    "&>div:not(:last-child)": {
      marginRight: `calc(${uikitStyles.theme.space.md} / 2)`
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
