// @hitachivantara/uikit-react-core/dist/cjs/components/TimePicker/Unit/Unit.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvTimePickerUnit", {
  root: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    textAlign: "center",
    ...uikitStyles.theme.typography.title3
  },
  separator: {
    width: 8
  },
  periodToggle: {
    height: 40,
    width: 40
  },
  element: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: 40,
    width: 40
  },
  input: {
    ...uikitStyles.theme.typography.title3,
    fontWeight: 600,
    textAlign: "center",
    height: 40,
    width: 40,
    padding: 0,
    margin: 0,
    "&::placeholder": {
      fontSize: 16,
      fontWeight: 600
    }
  },
  inputRoot: {
    width: 40,
    height: 40
  },
  subtractIcon: {
    marginTop: uikitStyles.theme.space.xs
  },
  inputContainer: {
    minWidth: 40,
    maxWidth: 40
  },
  inputBorderContainer: {
    top: 40
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
