// @hitachivantara/uikit-react-core/dist/cjs/components/Table/TableRow/TableRow.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const utils = require("../utils/utils.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvTableRow", {
  /** Styles applied to the component root class. */
  root: {
    color: "inherit",
    backgroundColor: uikitStyles.theme.table.rowBackgroundColor,
    verticalAlign: "middle",
    outline: 0
  },
  /** Styles applied to the component root when inside a `HvTableHead`. */
  head: {},
  /** Styles applied to the component root when inside a `HvTableBody`. */
  body: {},
  /** Styles applied to the component root when inside a `HvTableFooter`. */
  footer: {},
  /** Styles applied to the component root when selected. */
  selected: {
    backgroundColor: uikitStyles.theme.table.selectedRowBackgroundColor
  },
  /** Styles applied to the component root when expanded. */
  expanded: {
    backgroundColor: uikitStyles.theme.colors.atmo1,
    "& > *[role=cell]": {
      borderBottom: "none"
    }
  },
  /** Styles applied to the component root when striped. */
  striped: {},
  /** Styles applied to the component root on hover. */
  hover: {
    transition: "background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
    "&:hover": {
      backgroundColor: uikitStyles.theme.table.rowHoverColor
    }
  },
  /** Styles applied to the component root when its table variant is list. */
  variantList: {
    // only applied on custom `display`
    marginBottom: uikitStyles.theme.space.xs,
    borderRadius: uikitStyles.theme.table.rowListBorderRadius,
    ...utils.getBorderStyles("row", uikitStyles.theme.table.rowListBorderColor, uikitStyles.theme.table.rowListBorderRadius),
    backgroundColor: uikitStyles.theme.colors.atmo1,
    "&$selected": {
      ...utils.getBorderStyles("row", uikitStyles.theme.colors.secondary, uikitStyles.theme.table.rowListBorderRadius),
      "&:hover": {
        ...utils.getBorderStyles("row", uikitStyles.theme.table.rowHoverBorderColor, uikitStyles.theme.table.rowListBorderRadius)
      }
    },
    "&:hover": {
      ...utils.getBorderStyles("row", uikitStyles.theme.table.rowHoverBorderColor, uikitStyles.theme.table.rowListBorderRadius)
    },
    "&.HvIsFocused": {
      borderRadius: uikitStyles.theme.table.rowListBorderRadius
    }
  },
  /** Styles applied to the component root when its table variant is list. */
  variantListHead: {
    backgroundColor: uikitStyles.theme.table.rowListBackgroundColor
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
