// @hitachivantara/uikit-react-core/dist/cjs/components/Table/hooks/usePagination.cjs
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const reactTable = require("react-table");
const useInstanceHook = (instance) => {
  reactTable.ensurePluginOrder(instance.plugins, ["usePagination"], "useHvPagination");
  const getInstance = reactTable.useGetLatest(instance);
  const getHvPaginationProps = reactTable.makePropGetter(instance.getHooks().getHvPaginationProps, {
    instance: getInstance()
  });
  Object.assign(instance, {
    getHvPaginationProps
  });
};
const defaultGetHvPaginationProps = (props, {
  instance
}) => {
  const {
    canPreviousPage,
    canNextPage,
    pageOptions,
    gotoPage,
    setPageSize,
    state: {
      pageSize,
      pageIndex
    },
    labels
  } = instance;
  const nextProps = {
    canPrevious: canPreviousPage,
    canNext: canNextPage,
    pages: pageOptions.length,
    page: pageIndex,
    pageSize,
    onPageChange: gotoPage,
    onPageSizeChange: setPageSize,
    labels
  };
  return [props, nextProps];
};
const usePagination = (hooks) => {
  hooks.getHvPaginationProps = [defaultGetHvPaginationProps];
  hooks.useInstance.push(useInstanceHook);
};
usePagination.pluginName = "useHvPagination";
const usePagination$1 = usePagination;
exports.default = usePagination$1;
exports.defaultGetHvPaginationProps = defaultGetHvPaginationProps;
