// @hitachivantara/uikit-react-core/dist/cjs/components/BaseRadio/BaseRadio.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const MuiRadio = require("@mui/material/Radio");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const useDefaultProps = require("../../hooks/useDefaultProps.cjs");
const BaseRadio_styles = require("./BaseRadio.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const MuiRadio__default = /* @__PURE__ */ _interopDefault(MuiRadio);
const getSelectorIcons = (options, classes) => {
  const {
    disabled,
    semantic
  } = options;
  const color = disabled && ["atmo3", "secondary_60"] || semantic && ["base_light", "base_dark"] || void 0;
  const checkedColor = disabled && ["atmo3", "secondary_60"] || semantic && ["base_dark", "base_light"] || void 0;
  return {
    radio: /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.RadioButtonUnselected, { color, className: classes.icon }),
    radioChecked: /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.RadioButtonSelected, { color: checkedColor, className: classes.icon })
  };
};
const HvBaseRadio = (props) => {
  const {
    classes: classesProp,
    className,
    id,
    name,
    value = "on",
    required = false,
    readOnly = false,
    disabled = false,
    checked,
    defaultChecked,
    onChange,
    semantic = false,
    inputProps,
    onFocusVisible,
    onBlur,
    ...others
  } = useDefaultProps.useDefaultProps("HvBaseRadio", props);
  const {
    classes,
    cx
  } = BaseRadio_styles.useClasses(classesProp);
  const [focusVisible, setFocusVisible] = React.useState(false);
  const onFocusVisibleCallback = React.useCallback((evt) => {
    setFocusVisible(true);
    onFocusVisible == null ? void 0 : onFocusVisible(evt);
  }, [onFocusVisible]);
  const onBlurCallback = React.useCallback((evt) => {
    setFocusVisible(false);
    onBlur == null ? void 0 : onBlur(evt);
  }, [onBlur]);
  const icons = getSelectorIcons({
    disabled,
    semantic
  }, classes);
  const onLocalChange = React.useCallback((evt) => {
    if (readOnly) {
      return;
    }
    onChange == null ? void 0 : onChange(evt, evt.target.checked, value);
  }, [onChange, readOnly, value]);
  return /* @__PURE__ */ jsxRuntime.jsx(MuiRadio__default.default, { id, name, className: cx(classes.root, {
    [classes.disabled]: disabled,
    [classes.focusVisible]: focusVisible
  }, className), icon: icons.radio, checkedIcon: icons.radioChecked, color: "default", disabled, required, readOnly, disableRipple: true, onChange: onLocalChange, value, checked, defaultChecked, inputProps, onFocusVisible: onFocusVisibleCallback, onBlur: onBlurCallback, ...others });
};
exports.baseRadioClasses = BaseRadio_styles.staticClasses;
exports.HvBaseRadio = HvBaseRadio;
exports.getSelectorIcons = getSelectorIcons;
