// @hitachivantara/uikit-react-core/dist/cjs/components/Header/Navigation/MenuBar/Bar.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../../utils/classes.cjs");
const show = {
  top: `calc(${uikitStyles.theme.header.height} - ${uikitStyles.theme.header.borderTopThickness})`,
  transition: ["top"],
  boxShadow: uikitStyles.theme.header.shadow,
  transitionDuration: "500ms",
  backgroundColor: uikitStyles.theme.header.secondLevelBackgroundColor
};
const hide = {
  top: 0,
  transition: ["top"],
  boxShadow: "none",
  transitionDuration: "300ms"
};
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvHeader-MenuBar", {
  root: {
    left: 0,
    width: "100%",
    height: "100%",
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  },
  menu: {
    position: "absolute",
    zIndex: -2,
    height: uikitStyles.theme.header.secondLevelHeight,
    backgroundColor: uikitStyles.theme.header.secondLevelBackgroundColor,
    "& li > div": {
      marginTop: 0
    }
  },
  menubar: {
    position: "relative",
    backgroundColor: uikitStyles.theme.header.backgroundColor
  },
  hidden: {
    ...hide
  },
  active: {
    ...show
  },
  list: {
    margin: 0,
    padding: 0,
    display: "inherit",
    alignItems: "center",
    height: "100%",
    "&:hover $active": {
      ...hide
    },
    "& li:hover > $hidden": {
      ...show
    },
    "&:focus-within $active": {
      ...hide,
      zIndex: -2
    },
    "& li:focus-within > $hidden": {
      ...show,
      zIndex: -1
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
