// @hitachivantara/uikit-react-core/dist/cjs/components/BulkActions/BulkActions.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvBulkActions", {
  root: {
    display: "flex",
    alignItems: "center",
    border: uikitStyles.theme.bulkActions.border,
    backgroundColor: uikitStyles.theme.bulkActions.backgroundColor,
    padding: uikitStyles.theme.bulkActions.padding,
    marginBottom: uikitStyles.theme.space.xs
  },
  semantic: {
    backgroundColor: uikitStyles.theme.bulkActions.anySelectedBackgroundColor
  },
  actions: {
    display: "inline-flex",
    marginLeft: "auto"
  },
  selectAllContainer: {
    display: "flex",
    alignItems: "center"
  },
  selectAll: {},
  selectAllPages: {},
  divider: {
    display: uikitStyles.theme.bulkActions.separatorDisplay,
    backgroundColor: uikitStyles.theme.colors.atmo4,
    width: "1px",
    height: "32px",
    marginLeft: uikitStyles.theme.space.sm
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
