// @hitachivantara/uikit-react-core/dist/cjs/components/Dialog/Dialog.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvDialog", {
  root: {},
  background: {},
  paper: {
    color: uikitStyles.theme.colors.secondary,
    backgroundColor: uikitStyles.theme.colors.atmo1,
    boxShadow: ["none", uikitStyles.theme.colors.shadow],
    borderRadius: uikitStyles.theme.dialog.borderRadius
  },
  fullscreen: {},
  closeButton: {
    padding: 0,
    minWidth: "auto",
    position: "absolute",
    top: uikitStyles.theme.space.sm,
    right: uikitStyles.theme.space.sm,
    width: 32,
    height: 32
  },
  statusBar: {
    borderTopLeftRadius: 0,
    borderTopRightRadius: 0,
    borderTopWidth: 4,
    borderTopStyle: "solid"
  },
  success: {
    borderTopColor: uikitStyles.theme.colors.positive
  },
  error: {
    borderTopColor: uikitStyles.theme.colors.negative
  },
  warning: {
    borderTopColor: uikitStyles.theme.colors.warning
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
