// @hitachivantara/uikit-react-core/dist/cjs/components/Calendar/CalendarNavigation/MonthSelector/MonthSelector.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../../utils/focusUtils.cjs");
const classes = require("../../../../utils/classes.cjs");
const hover = {
  backgroundColor: uikitStyles.theme.colors.atmo3,
  cursor: "pointer"
};
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvMonthSelector", {
  calendarMonthlyGrid: {
    marginTop: "50px",
    marginLeft: "-16px",
    display: "flex",
    zIndex: "10",
    padding: "0 20px",
    position: "absolute",
    flexFlow: "wrap",
    alignContent: "center",
    justifyContent: "space-evenly",
    backgroundColor: uikitStyles.theme.colors.atmo1
  },
  rangeModeWidth: {},
  normalWidth: {},
  focusSelection: {
    "&:hover": {
      ...hover
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      ...hover,
      ...focusUtils.outlineStyles
    }
  },
  calendarMonthlyCell: {
    display: "flex",
    justifyContent: "center",
    flexDirection: "column",
    textAlign: "center",
    height: "40px",
    width: "92px",
    "&:hover": {
      ...hover
    }
  },
  calendarMonthlyCellSelected: {
    backgroundColor: uikitStyles.theme.colors.atmo3,
    color: uikitStyles.theme.colors.secondary,
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      color: uikitStyles.theme.colors.secondary
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
