// @hitachivantara/uikit-react-core/dist/cjs/components/ScrollTo/Horizontal/HorizontalScrollListItem/HorizontalScrollListItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../../utils/focusUtils.cjs");
const classes = require("../../../../utils/classes.cjs");
const name = "HvHorizontalScrollListItem";
const {
  staticClasses,
  useClasses
} = classes.createClasses(name, {
  root: {
    padding: "10px 0"
  },
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: uikitStyles.theme.scrollTo.horizontal.buttonHeight,
    cursor: "pointer",
    borderBottom: uikitStyles.theme.scrollTo.horizontal.buttonBottomBorder,
    "&:hover": {
      backgroundColor: uikitStyles.theme.scrollTo.horizontal.buttonHoverBackgroundColor,
      "& $notSelected": {
        height: uikitStyles.theme.scrollTo.dotHoverSize,
        width: uikitStyles.theme.scrollTo.dotHoverSize,
        backgroundColor: uikitStyles.theme.scrollTo.dotHoverColor
      },
      "& $notSelectedRoot": {
        backgroundColor: uikitStyles.theme.scrollTo.dotHoverBackgroundColor
      }
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    }
  },
  text: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    height: uikitStyles.theme.scrollTo.horizontal.buttonHeight,
    borderBottom: uikitStyles.theme.scrollTo.horizontal.buttonBottomBorder,
    "& p": {
      padding: uikitStyles.theme.scrollTo.horizontal.textPadding,
      maxWidth: uikitStyles.theme.scrollTo.horizontal.textMaxWidth,
      textOverflow: "ellipsis",
      overflow: "hidden"
    }
  },
  selected: {
    borderBottom: uikitStyles.theme.scrollTo.horizontal.selectedButtonBottomBorder
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
