// @hitachivantara/uikit-react-core/dist/cjs/components/AppSwitcher/Action/Action.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvAppSwitcher-Action", {
  root: {
    width: "100%",
    maxWidth: 280,
    minHeight: 52,
    marginRight: uikitStyles.theme.space.sm
  },
  icon: {
    display: "flex",
    minWidth: 40,
    justifyContent: "center"
  },
  iconUrl: {
    width: 32
  },
  iconInfo: {
    minWidth: 32
  },
  disabled: {},
  selected: {},
  typography: {
    display: "flex",
    justifyContent: "flex-start",
    alignItems: "center",
    width: "100%",
    minHeight: 52,
    padding: `6px ${uikitStyles.theme.space.xs}`,
    border: "none",
    borderLeft: `solid 2px ${uikitStyles.theme.colors.secondary}`,
    cursor: "pointer",
    textDecoration: "inherit",
    color: "inherit",
    backgroundColor: "inherit",
    "$disabled &": {
      cursor: "not-allowed"
    }
  },
  title: {
    flexGrow: 1,
    margin: `0 ${uikitStyles.theme.space.xs}`,
    textAlign: "left",
    overflow: "hidden",
    whiteSpace: "nowrap",
    textOverflow: "ellipsis",
    color: "inherit"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
