// @hitachivantara/uikit-react-core/dist/esm/components/FileUploader/Preview/Preview.js
import { jsxs, jsx } from "@emotion/react/jsx-runtime";
import { useEffect } from "react";
import { Preview } from "@hitachivantara/uikit-react-icons";
import { useClasses } from "./Preview.styles.js";
import { staticClasses } from "./Preview.styles.js";
import { HvButton } from "../../Button/Button.js";
const HvFileUploaderPreview = ({
  className,
  children,
  classes: classesProp,
  disableOverlay = false,
  onUnload,
  onClick,
  ...others
}) => {
  const {
    classes,
    cx,
    css
  } = useClasses(classesProp);
  useEffect(() => {
    return () => {
      onUnload == null ? void 0 : onUnload();
    };
  }, [onUnload]);
  if (onClick) {
    return /* @__PURE__ */ jsxs(HvButton, { icon: true, className: cx(classes.previewButton, className), onClick, ...others, children: [
      children,
      !disableOverlay && /* @__PURE__ */ jsx("div", { className: classes.overlay, "aria-hidden": "true", children: /* @__PURE__ */ jsx(Preview, { className: css({
        position: "absolute",
        left: "50%",
        top: "50%",
        transform: "translate(-50%, -50%)",
        display: "none"
      }) }) })
    ] });
  }
  return children;
};
export {
  HvFileUploaderPreview,
  staticClasses as fileUploaderPreviewClasses
};
