// @hitachivantara/uikit-react-core/dist/cjs/components/Banner/BannerContent/BannerContent.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const classes = require("../../../utils/classes.cjs");
const {
  useClasses,
  staticClasses
} = classes.createClasses("HvBannerContent", {
  root: {
    minWidth: "100%",
    position: "relative",
    display: "flex",
    flexWrap: "nowrap",
    boxShadow: "none",
    borderRadius: uikitStyles.theme.radii.none
  },
  message: {
    display: "flex",
    alignItems: "center",
    padding: uikitStyles.theme.spacing("xs", 0),
    paddingLeft: uikitStyles.theme.space.sm
  },
  action: {
    padding: uikitStyles.theme.spacing("xs"),
    marginRight: 0,
    flex: "0 0 auto",
    placeSelf: "stretch"
  },
  baseVariant: {
    padding: 0
  },
  success: {
    backgroundColor: uikitStyles.theme.colors.positive_20
  },
  warning: {
    backgroundColor: uikitStyles.theme.colors.warning_20
  },
  error: {
    backgroundColor: uikitStyles.theme.colors.negative_20
  },
  default: {
    backgroundColor: uikitStyles.theme.colors.neutral_20
  },
  outContainer: {
    width: "100%",
    position: "relative"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
