// @hitachivantara/uikit-react-core/dist/cjs/components/Banner/BannerContent/BannerContent.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("@emotion/react/jsx-runtime");
const React = require("react");
const SnackbarContent = require("@mui/material/SnackbarContent");
const iconVariant = require("../../../utils/iconVariant.cjs");
const BannerContent_styles = require("./BannerContent.styles.cjs");
const MessageContainer = require("./MessageContainer/MessageContainer.cjs");
const ActionContainer = require("./ActionContainer/ActionContainer.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const SnackbarContent__default = /* @__PURE__ */ _interopDefault(SnackbarContent);
const HvBannerContent = React.forwardRef((props, ref) => {
  const {
    id,
    classes: classesProp,
    showIcon = false,
    customIcon,
    variant = "default",
    onClose,
    actions,
    actionsCallback,
    actionsPosition = "auto",
    content,
    actionProps,
    ...others
  } = props;
  const {
    classes,
    cx
  } = BannerContent_styles.useClasses(classesProp);
  const icon = customIcon || showIcon && iconVariant.iconVariant(variant, "base_dark");
  const effectiveActionsPosition = actionsPosition === "auto" ? "inline" : actionsPosition;
  return /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.outContainer, children: /* @__PURE__ */ jsxRuntime.jsx(SnackbarContent__default.default, { ref, id, classes: {
    root: classes.root,
    message: classes.message,
    action: classes.action
  }, className: cx(classes == null ? void 0 : classes.baseVariant, classes == null ? void 0 : classes[variant]), message: /* @__PURE__ */ jsxRuntime.jsx(MessageContainer.HvMessageContainer, { id, icon, ...effectiveActionsPosition === "inline" && {
    actionsOnMessage: actions,
    actionsOnMessageCallback: actionsCallback
  }, message: content }), action: /* @__PURE__ */ jsxRuntime.jsx(ActionContainer.HvActionContainer, { id, onClose, ...effectiveActionsPosition === "bottom-right" && {
    action: actions,
    actionCallback: actionsCallback
  }, ...actionProps }), ...others }) });
});
exports.bannerContentClasses = BannerContent_styles.staticClasses;
exports.HvBannerContent = HvBannerContent;
