// @hitachivantara/uikit-react-core/dist/cjs/components/Calendar/CalendarNavigation/Navigation/Navigation.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../../../utils/focusUtils.cjs");
const classes = require("../../../../utils/classes.cjs");
const {
  staticClasses,
  useClasses
} = classes.createClasses("HvNavigation", {
  root: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between"
  },
  icon: {
    userSelect: "none",
    width: "32px",
    height: "32px",
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      cursor: "pointer"
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      cursor: "pointer",
      ...focusUtils.outlineStyles
    }
  },
  disabled: {},
  text: {
    width: "calc(100% - 64px)",
    textAlign: "center",
    height: "32px",
    padding: "8px 0",
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      cursor: "pointer"
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      backgroundColor: uikitStyles.theme.colors.atmo3,
      cursor: "pointer",
      ...focusUtils.outlineStyles
    }
  },
  textWithoutHover: {
    width: "calc(100% - 64px)",
    textAlign: "center",
    padding: "0 5px",
    outline: "none"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
